{
    "items": [
        {
            "tags": [
                "r",
                "loops",
                "dataframe"
            ],
            "owner": {
                "reputation": 296,
                "user_id": 7974647,
                "user_type": "registered",
                "accept_rate": 78,
                "profile_image": "https://www.gravatar.com/avatar/06eec443f6b2c658c78e8491a489a4b4?s=128&d=identicon&r=PG&f=1",
                "display_name": "Kay",
                "link": "https://stackoverflow.com/users/7974647/kay"
            },
            "is_answered": false,
            "view_count": 34,
            "answer_count": 0,
            "score": 1,
            "last_activity_date": 1524185886,
            "creation_date": 1524185886,
            "question_id": 49932314,
            "body_markdown": "I have 3 df which I put them in a list. For each df, I adjust values in a `zvar` function. Then I want to run a model and then resampling for the model in a loop applying the `zvar`. \r\n\r\nBut I encounter this error:\r\n\r\n&gt;Error in `$&lt;-.data.frame`(`*tmp*`, &quot;var2&quot;, value = numeric(0)) : \r\n  replacement has 0 rows, data has 200\r\n\r\nThe message says that a variable does not exist, but I am able to print each variable in the dataframe, what is the issue here?\r\n\r\nMy df sample (all continuous vars):\r\n\r\n      var1     var2\r\n       2        2\r\n       1        2\r\n       9        7\r\n\r\nMy code:\r\n\r\n    zvar &lt;- function(data, mean, sd){\r\n         data$var2 &lt;- (data$var2 - mean$var2)/sd$var2\r\n         data$var1 &lt;- (data$var1 - mean$var1)/sd$var1\r\n         return(data)\r\n     }\r\n     # its args are automatically calculated in the following function `boot.fiml`.\r\n     \r\n\r\n    dflist &lt;- list(group1, group2, group3)\r\n\r\n    for (i in dflist) {\r\n       i &lt;- data.frame(i)\r\n       print(i$var2)\r\n       print(i$var1)\r\n       m_group &lt;- lm(var1 ~ var2, data = i)\r\n       print(m_group)\r\n       \r\n       ## previous lines run successfully, not not the following one\r\n       boot.fiml(data = i, model = m_group, R = 10, z.function = zvar)\r\n     }\r\n\r\nThanks!",
            "link": "https://stackoverflow.com/questions/49932314/how-to-solve-error-replacement-has-0-rows-data-has-200-for-a-loop",
            "title": "how to solve error replacement has 0 rows, data has 200 for a loop",
            "body": "<p>I have 3 df which I put them in a list. For each df, I adjust values in a <code>zvar</code> function. Then I want to run a model and then resampling for the model in a loop applying the <code>zvar</code>. </p>\n\n<p>But I encounter this error:</p>\n\n<blockquote>\n  <p>Error in <code>$&lt;-.data.frame</code>(<code>*tmp*</code>, \"var2\", value = numeric(0)) : \n    replacement has 0 rows, data has 200</p>\n</blockquote>\n\n<p>The message says that a variable does not exist, but I am able to print each variable in the dataframe, what is the issue here?</p>\n\n<p>My df sample (all continuous vars):</p>\n\n<pre><code>  var1     var2\n   2        2\n   1        2\n   9        7\n</code></pre>\n\n<p>My code:</p>\n\n<pre><code>zvar &lt;- function(data, mean, sd){\n     data$var2 &lt;- (data$var2 - mean$var2)/sd$var2\n     data$var1 &lt;- (data$var1 - mean$var1)/sd$var1\n     return(data)\n }\n # its args are automatically calculated in the following function `boot.fiml`.\n\n\ndflist &lt;- list(group1, group2, group3)\n\nfor (i in dflist) {\n   i &lt;- data.frame(i)\n   print(i$var2)\n   print(i$var1)\n   m_group &lt;- lm(var1 ~ var2, data = i)\n   print(m_group)\n\n   ## previous lines run successfully, not not the following one\n   boot.fiml(data = i, model = m_group, R = 10, z.function = zvar)\n }\n</code></pre>\n\n<p>Thanks!</p>\n"
        },
        {
            "tags": [
                "java",
                "maven",
                "junit4",
                "mockito",
                "powermock"
            ],
            "owner": {
                "reputation": 1042,
                "user_id": 1521346,
                "user_type": "registered",
                "accept_rate": 47,
                "profile_image": "https://i.stack.imgur.com/VHHCo.jpg?s=128&g=1",
                "display_name": "Bhuvan",
                "link": "https://stackoverflow.com/users/1521346/bhuvan"
            },
            "is_answered": true,
            "view_count": 15481,
            "accepted_answer_id": 13043338,
            "answer_count": 4,
            "score": 7,
            "last_activity_date": 1524185885,
            "creation_date": 1346304321,
            "last_edit_date": 1524185885,
            "question_id": 12190377,
            "body_markdown": "I have integrated PowerMock and PowerRule in JUnit with Mockito.\r\n\r\nHere are my dependencies:\r\n\r\n    &lt;dependency&gt;\r\n            &lt;groupId&gt;javassist&lt;/groupId&gt;\r\n            &lt;artifactId&gt;javassist&lt;/artifactId&gt;\r\n            &lt;version&gt;3.12.0.GA&lt;/version&gt;\r\n          &lt;/dependency&gt;\r\n    &lt;dependency&gt;\r\n\t\t\t&lt;groupId&gt;asm&lt;/groupId&gt;\r\n\t\t\t&lt;artifactId&gt;asm&lt;/artifactId&gt;\r\n\t\t\t&lt;version&gt;3.3.1&lt;/version&gt;\r\n\t&lt;/dependency&gt;\r\n\r\n     &lt;dependency&gt;\r\n\t        &lt;groupId&gt;cglib&lt;/groupId&gt;\r\n\t        &lt;artifactId&gt;cglib&lt;/artifactId&gt;\r\n\t        &lt;version&gt;2.2.2&lt;/version&gt;\r\n     &lt;/dependency&gt;\r\n     &lt;dependency&gt;\r\n\t        &lt;groupId&gt;org.powermoc&lt;/groupId&gt;\r\n\t\t&lt;artifactId&gt;powermock-module-junit4&lt;/artifactId&gt;\r\n\t\t&lt;version&gt;1.4.12&lt;/version&gt;\r\n\t\t&lt;scope&gt;test&lt;/scope&gt;\r\n\t\t&lt;/dependency&gt;\r\n     &lt;dependency&gt;\r\n\t\t&lt;groupId&gt;org.powermock&lt;/groupId&gt;\r\n\t\t&lt;artifactId&gt;powermock-api-mockito&lt;/artifactId&gt;\r\n\t\t&lt;version&gt;1.4.12&lt;/version&gt;\r\n\t\t&lt;scope&gt;test&lt;/scope&gt;\r\n     &lt;/dependency&gt;\r\n     &lt;dependency&gt;\r\n\t       &lt;groupId&gt;org.powermock&lt;/groupId&gt;\r\n\t       &lt;artifactId&gt;powermock-module-junit4-rule&lt;/artifactId&gt;\r\n\t       &lt;version&gt;1.4.12&lt;/version&gt;\r\n\t       &lt;scope&gt;test&lt;/scope&gt;\r\n     &lt;/dependency&gt;\r\n     &lt;dependency&gt;\r\n\t        &lt;groupId&gt;org.powermock&lt;/groupId&gt;\r\n\t        &lt;artifactId&gt;powermock-classloading-objenesis&lt;/artifactId&gt;\r\n\t        &lt;version&gt;1.4.12&lt;/version&gt;\r\n\t        &lt;scope&gt;test&lt;/scope&gt;\r\n     &lt;/dependency&gt;\r\n\r\nMy test class is: \r\n\r\n    public class TestClass extends AbstractShiroTest{\r\n        @Rule\r\n        public PowerMockRule rule = new PowerMockRule();\r\n\t    @Autowired\r\n\t    SomeService someService;\r\n        @Before\r\n\r\n\t    public void setUp(){\r\n\t\tMap&lt;String, Object&gt; newMap = new HashMap&lt;String, Object&gt;();\r\n\t\tnewMap.put(&quot;userTimeZone&quot;, &quot;Asia/Calcutta&quot;);\r\n\t\tSubject subjectUnderTest = mock(Subject.class);\r\n\t\t         when(subjectUnderTest.getPrincipal()).thenReturn(LMPTestConstants.USER_NAME);\r\n\t\tSession session = mock(Session.class);\r\n\t\twhen(session.getAttribute(LMPCoreConstants.USER_DETAILS_MAP)).thenReturn(newMap);\r\n\t\twhen(subjectUnderTest.getSession(false)).thenReturn(session);\r\n      \tsetSubject(subjectUnderTest);\r\n      \tPowerMockito.mockStatic(CasSessionUtil.class);\r\n    \twhen(CasSessionUtil.getCarrierId()).thenReturn(1L);\r\n\t}\r\n\r\n      @Test\r\n\r\n    public void myTestMethod()  {\r\n    someService.doSomething();\r\n     }\r\n    }\r\n\r\n**`doSomething` is calling a static method that I need to mock.\r\nWhen I run my test case, I get `javassist.NotFoundException: $Proxy88`.**\r\n\r\nFull stack trace:\r\n\r\n&gt; java.lang.RuntimeException: javassist.NotFoundException: $Proxy88\r\n\tat org.powermock.core.classloader.MockClassLoader.loadUnmockedClass(MockClassLoader.java:187)\r\n\tat org.powermock.core.classloader.MockClassLoader.loadModifiedClass(MockClassLoader.java:147)\r\n\tat org.powermock.core.classloader.DeferSupportingClassLoader.loadClass(DeferSupportingClassLoader.java:67)\r\n\tat java.lang.ClassLoader.loadClass(ClassLoader.java:252)\r\n\tat java.lang.ClassLoader.loadClassInternal(ClassLoader.java:320)\r\n\tat java.lang.Class.forName0(Native Method)\r\n\tat java.lang.Class.forName(Class.java:247)\r\n\tat org.powermock.api.support.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:66)\r\n\tat org.powermock.api.support.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:26)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:243)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\r\n\tat org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\r\n\tat org.powermock.classloading.DeepCloner.clone(DeepCloner.java:82)\r\n\tat org.powermock.classloading.DeepCloner.clone(DeepCloner.java:69)\r\n\tat org.powermock.classloading.ClassloaderExecutor.execute(ClassloaderExecutor.java:89)\r\n\tat org.powermock.classloading.ClassloaderExecutor.execute(ClassloaderExecutor.java:78)\r\n\tat org.powermock.modules.junit4.rule.PowerMockStatement.evaluate(PowerMockRule.java:49)\r\n\tat org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:72)\r\n\tat org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:231)\r\n\tat org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)\r\n\tat org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)\r\n\tat org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)\r\n\tat org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)\r\n\tat org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)\r\n\tat org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)\r\n\tat org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)\r\n\tat org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)\r\n\tat org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:71)\r\n\tat org.junit.runners.ParentRunner.run(ParentRunner.java:236)\r\n\tat org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:174)\r\n\tat org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)\r\n\tat org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)\r\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:467)\r\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:683)\r\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:390)\r\n\tat org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:197)\r\nCaused by: javassist.NotFoundException: $Proxy88\r\n\tat javassist.ClassPool.get(ClassPool.java:436)\r\n\tat org.powermock.core.classloader.MockClassLoader.loadUnmockedClass(MockClassLoader.java:180)\r\n\t... 46 more\r\n\r\nIf I change the dependency to\r\n\r\n    &lt;dependency&gt;\r\n    \t        &lt;groupId&gt;org.powermock&lt;/groupId&gt;\r\n    \t        &lt;artifactId&gt;powermock-classloading-xstream&lt;/artifactId&gt;\r\n    \t        &lt;version&gt;1.4.12&lt;/version&gt;\r\n    \t        &lt;scope&gt;test&lt;/scope&gt;\r\n    \t\t&lt;/dependency&gt;\r\n\r\nthen I get a different exception. Please see https://stackoverflow.com/questions/12176049/suggest-work-around-for-com-thoughtworks-xstream-converters-conversionexception (deleted SO question, requires 10k).\r\n\r\nI have even tried with javassist version 15, but that has the same problem.",
            "link": "https://stackoverflow.com/questions/12190377/getting-javassist-notfoundexception-with-powermock-and-powerrule-in-junit-with-m",
            "title": "Getting javassist.NotFoundException with PowerMock and PowerRule in JUnit with Mockito",
            "body": "<p>I have integrated PowerMock and PowerRule in JUnit with Mockito.</p>\n\n<p>Here are my dependencies:</p>\n\n<pre><code>&lt;dependency&gt;\n        &lt;groupId&gt;javassist&lt;/groupId&gt;\n        &lt;artifactId&gt;javassist&lt;/artifactId&gt;\n        &lt;version&gt;3.12.0.GA&lt;/version&gt;\n      &lt;/dependency&gt;\n&lt;dependency&gt;\n        &lt;groupId&gt;asm&lt;/groupId&gt;\n        &lt;artifactId&gt;asm&lt;/artifactId&gt;\n        &lt;version&gt;3.3.1&lt;/version&gt;\n&lt;/dependency&gt;\n\n &lt;dependency&gt;\n        &lt;groupId&gt;cglib&lt;/groupId&gt;\n        &lt;artifactId&gt;cglib&lt;/artifactId&gt;\n        &lt;version&gt;2.2.2&lt;/version&gt;\n &lt;/dependency&gt;\n &lt;dependency&gt;\n        &lt;groupId&gt;org.powermoc&lt;/groupId&gt;\n    &lt;artifactId&gt;powermock-module-junit4&lt;/artifactId&gt;\n    &lt;version&gt;1.4.12&lt;/version&gt;\n    &lt;scope&gt;test&lt;/scope&gt;\n    &lt;/dependency&gt;\n &lt;dependency&gt;\n    &lt;groupId&gt;org.powermock&lt;/groupId&gt;\n    &lt;artifactId&gt;powermock-api-mockito&lt;/artifactId&gt;\n    &lt;version&gt;1.4.12&lt;/version&gt;\n    &lt;scope&gt;test&lt;/scope&gt;\n &lt;/dependency&gt;\n &lt;dependency&gt;\n       &lt;groupId&gt;org.powermock&lt;/groupId&gt;\n       &lt;artifactId&gt;powermock-module-junit4-rule&lt;/artifactId&gt;\n       &lt;version&gt;1.4.12&lt;/version&gt;\n       &lt;scope&gt;test&lt;/scope&gt;\n &lt;/dependency&gt;\n &lt;dependency&gt;\n        &lt;groupId&gt;org.powermock&lt;/groupId&gt;\n        &lt;artifactId&gt;powermock-classloading-objenesis&lt;/artifactId&gt;\n        &lt;version&gt;1.4.12&lt;/version&gt;\n        &lt;scope&gt;test&lt;/scope&gt;\n &lt;/dependency&gt;\n</code></pre>\n\n<p>My test class is: </p>\n\n<pre><code>public class TestClass extends AbstractShiroTest{\n    @Rule\n    public PowerMockRule rule = new PowerMockRule();\n    @Autowired\n    SomeService someService;\n    @Before\n\n    public void setUp(){\n    Map&lt;String, Object&gt; newMap = new HashMap&lt;String, Object&gt;();\n    newMap.put(\"userTimeZone\", \"Asia/Calcutta\");\n    Subject subjectUnderTest = mock(Subject.class);\n             when(subjectUnderTest.getPrincipal()).thenReturn(LMPTestConstants.USER_NAME);\n    Session session = mock(Session.class);\n    when(session.getAttribute(LMPCoreConstants.USER_DETAILS_MAP)).thenReturn(newMap);\n    when(subjectUnderTest.getSession(false)).thenReturn(session);\n    setSubject(subjectUnderTest);\n    PowerMockito.mockStatic(CasSessionUtil.class);\n    when(CasSessionUtil.getCarrierId()).thenReturn(1L);\n}\n\n  @Test\n\npublic void myTestMethod()  {\nsomeService.doSomething();\n }\n}\n</code></pre>\n\n<p><strong><code>doSomething</code> is calling a static method that I need to mock.\nWhen I run my test case, I get <code>javassist.NotFoundException: $Proxy88</code>.</strong></p>\n\n<p>Full stack trace:</p>\n\n<blockquote>\n  <p>java.lang.RuntimeException: javassist.NotFoundException: $Proxy88\n      at org.powermock.core.classloader.MockClassLoader.loadUnmockedClass(MockClassLoader.java:187)\n      at org.powermock.core.classloader.MockClassLoader.loadModifiedClass(MockClassLoader.java:147)\n      at org.powermock.core.classloader.DeferSupportingClassLoader.loadClass(DeferSupportingClassLoader.java:67)\n      at java.lang.ClassLoader.loadClass(ClassLoader.java:252)\n      at java.lang.ClassLoader.loadClassInternal(ClassLoader.java:320)\n      at java.lang.Class.forName0(Native Method)\n      at java.lang.Class.forName(Class.java:247)\n      at org.powermock.api.support.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:66)\n      at org.powermock.api.support.ClassLoaderUtil.loadClass(ClassLoaderUtil.java:26)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:243)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.cloneFields(DeepCloner.java:248)\n      at org.powermock.classloading.DeepCloner.performClone(DeepCloner.java:128)\n      at org.powermock.classloading.DeepCloner.clone(DeepCloner.java:82)\n      at org.powermock.classloading.DeepCloner.clone(DeepCloner.java:69)\n      at org.powermock.classloading.ClassloaderExecutor.execute(ClassloaderExecutor.java:89)\n      at org.powermock.classloading.ClassloaderExecutor.execute(ClassloaderExecutor.java:78)\n      at org.powermock.modules.junit4.rule.PowerMockStatement.evaluate(PowerMockRule.java:49)\n      at org.springframework.test.context.junit4.statements.SpringRepeat.evaluate(SpringRepeat.java:72)\n      at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.runChild(SpringJUnit4ClassRunner.java:231)\n      at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:50)\n      at org.junit.runners.ParentRunner$3.run(ParentRunner.java:193)\n      at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:52)\n      at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:191)\n      at org.junit.runners.ParentRunner.access$000(ParentRunner.java:42)\n      at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:184)\n      at org.springframework.test.context.junit4.statements.RunBeforeTestClassCallbacks.evaluate(RunBeforeTestClassCallbacks.java:61)\n      at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:31)\n      at org.springframework.test.context.junit4.statements.RunAfterTestClassCallbacks.evaluate(RunAfterTestClassCallbacks.java:71)\n      at org.junit.runners.ParentRunner.run(ParentRunner.java:236)\n      at org.springframework.test.context.junit4.SpringJUnit4ClassRunner.run(SpringJUnit4ClassRunner.java:174)\n      at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:50)\n      at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:38)\n      at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:467)\n      at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:683)\n      at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:390)\n      at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:197)\n  Caused by: javassist.NotFoundException: $Proxy88\n      at javassist.ClassPool.get(ClassPool.java:436)\n      at org.powermock.core.classloader.MockClassLoader.loadUnmockedClass(MockClassLoader.java:180)\n      ... 46 more</p>\n</blockquote>\n\n<p>If I change the dependency to</p>\n\n<pre><code>&lt;dependency&gt;\n            &lt;groupId&gt;org.powermock&lt;/groupId&gt;\n            &lt;artifactId&gt;powermock-classloading-xstream&lt;/artifactId&gt;\n            &lt;version&gt;1.4.12&lt;/version&gt;\n            &lt;scope&gt;test&lt;/scope&gt;\n        &lt;/dependency&gt;\n</code></pre>\n\n<p>then I get a different exception. Please see <a href=\"https://stackoverflow.com/questions/12176049/suggest-work-around-for-com-thoughtworks-xstream-converters-conversionexception\">https://stackoverflow.com/questions/12176049/suggest-work-around-for-com-thoughtworks-xstream-converters-conversionexception</a> (deleted SO question, requires 10k).</p>\n\n<p>I have even tried with javassist version 15, but that has the same problem.</p>\n"
        },
        {
            "tags": [
                "r",
                "performance",
                "group-by",
                "data.table"
            ],
            "owner": {
                "reputation": 7351,
                "user_id": 714319,
                "user_type": "registered",
                "accept_rate": 90,
                "profile_image": "https://www.gravatar.com/avatar/8da9ec11222ad8324ee8bafb392111e6?s=128&d=identicon&r=PG",
                "display_name": "Alex",
                "link": "https://stackoverflow.com/users/714319/alex"
            },
            "is_answered": true,
            "view_count": 9203,
            "accepted_answer_id": 27004566,
            "answer_count": 1,
            "score": 50,
            "last_activity_date": 1524185873,
            "creation_date": 1416344618,
            "last_edit_date": 1524185873,
            "question_id": 27004002,
            "body_markdown": "I cannot seem to find any documentation on what exactly `.EACHI` does in `data.table`. I see a brief mention of it in the documentation:\r\n\r\n&gt; Aggregation for a subset of known groups is particularly efficient\r\n&gt; when passing those groups in i and setting `by=.EACHI`. When `i` is a\r\n&gt; data.table, `DT[i,j,by=.EACHI]` evaluates `j` for the groups of `DT` that\r\n&gt; each row in `i` joins to. We call this grouping by each i.\r\n\r\nBut what does &quot;groups&quot; in the context of `DT` mean? Is a group determined by the key that is set on `DT`? Is the group every distinct row that uses all the columns as the key? I fully understand how to run something like `DT[i,j,by=my_grouping_variable]` but am confused as to how `.EACHI` would work. Could someone explain please?",
            "link": "https://stackoverflow.com/questions/27004002/eachi-in-data-table",
            "title": ".EACHI in data.table?",
            "body": "<p>I cannot seem to find any documentation on what exactly <code>.EACHI</code> does in <code>data.table</code>. I see a brief mention of it in the documentation:</p>\n\n<blockquote>\n  <p>Aggregation for a subset of known groups is particularly efficient\n  when passing those groups in i and setting <code>by=.EACHI</code>. When <code>i</code> is a\n  data.table, <code>DT[i,j,by=.EACHI]</code> evaluates <code>j</code> for the groups of <code>DT</code> that\n  each row in <code>i</code> joins to. We call this grouping by each i.</p>\n</blockquote>\n\n<p>But what does \"groups\" in the context of <code>DT</code> mean? Is a group determined by the key that is set on <code>DT</code>? Is the group every distinct row that uses all the columns as the key? I fully understand how to run something like <code>DT[i,j,by=my_grouping_variable]</code> but am confused as to how <code>.EACHI</code> would work. Could someone explain please?</p>\n"
        },
        {
            "tags": [
                "sql",
                "postgresql",
                "tree"
            ],
            "owner": {
                "reputation": 709,
                "user_id": 1500623,
                "user_type": "registered",
                "accept_rate": 79,
                "profile_image": "https://www.gravatar.com/avatar/6d7311019c952e2be3abfe71b16f243f?s=128&d=identicon&r=PG",
                "display_name": "Vladimir M.",
                "link": "https://stackoverflow.com/users/1500623/vladimir-m"
            },
            "is_answered": true,
            "view_count": 52,
            "accepted_answer_id": 49906005,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185872,
            "creation_date": 1523860129,
            "question_id": 49850810,
            "body_markdown": "Let&#39;s say I have a tree stored using ltree:\r\n\r\n       id   |   path   |   sort   \r\n    ------------------------------\r\n    0       |0         |1\r\n    1       |0.1       |2\r\n    2       |0.1.2     |3\r\n    3       |0.1.3     |1\r\n    4       |0.1.4     |2\r\n    5       |0.5       |3\r\n    6       |0.6       |1\r\n\r\nI want to select nodes so that:\r\n\r\n 1. child nodes appear right after the parent node;\r\n 2. sibling nodes that has lesser &quot;sort&quot; value appear first;\r\n\r\nLike this:\r\n\r\n       id   |   path   |   sort   \r\n    ------------------------------\r\n    0       |0         |1\r\n    6       |0.6       |1\r\n    1       |0.1       |2\r\n    3       |0.1.3     |1\r\n    4       |0.1.4     |2\r\n    2       |0.1.2     |3\r\n    5       |0.5       |3\r\n\r\nThe first requirement is possible with `ORDER BY path`, but I don&#39;t know how to implement the second, is this even possible?\r\n\r\n",
            "link": "https://stackoverflow.com/questions/49850810/sorting-nodes-with-postgresql-ltree",
            "title": "Sorting nodes with PostgreSQL ltree",
            "body": "<p>Let's say I have a tree stored using ltree:</p>\n\n<pre><code>   id   |   path   |   sort   \n------------------------------\n0       |0         |1\n1       |0.1       |2\n2       |0.1.2     |3\n3       |0.1.3     |1\n4       |0.1.4     |2\n5       |0.5       |3\n6       |0.6       |1\n</code></pre>\n\n<p>I want to select nodes so that:</p>\n\n<ol>\n<li>child nodes appear right after the parent node;</li>\n<li>sibling nodes that has lesser \"sort\" value appear first;</li>\n</ol>\n\n<p>Like this:</p>\n\n<pre><code>   id   |   path   |   sort   \n------------------------------\n0       |0         |1\n6       |0.6       |1\n1       |0.1       |2\n3       |0.1.3     |1\n4       |0.1.4     |2\n2       |0.1.2     |3\n5       |0.5       |3\n</code></pre>\n\n<p>The first requirement is possible with <code>ORDER BY path</code>, but I don't know how to implement the second, is this even possible?</p>\n"
        },
        {
            "tags": [
                "java",
                "r",
                "rjava"
            ],
            "owner": {
                "reputation": 113,
                "user_id": 7949644,
                "user_type": "registered",
                "profile_image": "https://i.stack.imgur.com/uzzcf.jpg?s=128&g=1",
                "display_name": "j.Rinehart",
                "link": "https://stackoverflow.com/users/7949644/j-rinehart"
            },
            "is_answered": true,
            "view_count": 2545,
            "accepted_answer_id": 43789178,
            "answer_count": 3,
            "score": 3,
            "last_activity_date": 1524185862,
            "creation_date": 1493917185,
            "last_edit_date": 1495540480,
            "question_id": 43788788,
            "body_markdown": "I am having a time with java related packages this week.  I finally got my RWeka working (see [HERE][1]), but now I&#39;m having rJava issues.  \r\n\r\nThe package seems to load up fine, but when I load the library I get the following:\r\n\r\n    Error in get(method, envir = home) : \r\n    lazy-load database &#39;/Library/Frameworks/R.framework/Versions/3.3/Resources/library/rJava/R/rJava.rdb&#39; is corrupt\r\n    In addition: Warning messages:\r\n    1: In .registerS3method(fin[i, 1], fin[i, 2], fin[i, 3], fin[i, 4],  :\r\n    restarting interrupted promise evaluation\r\n    2: In get(method, envir = home) :\r\n    restarting interrupted promise evaluation\r\n    3: In get(method, envir = home) : internal error -3 in R_decompress1\r\n    Error: package or namespace load failed for ‘rJava’\r\n\r\nAny ideas would be greatly appreciated!\r\n\r\n\r\n  [1]: https://stackoverflow.com/questions/43728983/mlr-fselector-install-error-java-related/43771982#43771982",
            "link": "https://stackoverflow.com/questions/43788788/another-rjava-library-install-error-rjava-rdb-is-corrupt",
            "title": "Another rJava library install error: rJava.rdb&#39; is corrupt",
            "body": "<p>I am having a time with java related packages this week.  I finally got my RWeka working (see <a href=\"https://stackoverflow.com/questions/43728983/mlr-fselector-install-error-java-related/43771982#43771982\">HERE</a>), but now I'm having rJava issues.  </p>\n\n<p>The package seems to load up fine, but when I load the library I get the following:</p>\n\n<pre><code>Error in get(method, envir = home) : \nlazy-load database '/Library/Frameworks/R.framework/Versions/3.3/Resources/library/rJava/R/rJava.rdb' is corrupt\nIn addition: Warning messages:\n1: In .registerS3method(fin[i, 1], fin[i, 2], fin[i, 3], fin[i, 4],  :\nrestarting interrupted promise evaluation\n2: In get(method, envir = home) :\nrestarting interrupted promise evaluation\n3: In get(method, envir = home) : internal error -3 in R_decompress1\nError: package or namespace load failed for ‘rJava’\n</code></pre>\n\n<p>Any ideas would be greatly appreciated!</p>\n"
        },
        {
            "tags": [
                "apache-spark",
                "pyspark",
                "apache-spark-mllib",
                "apache-spark-ml"
            ],
            "owner": {
                "reputation": 87,
                "user_id": 2705212,
                "user_type": "registered",
                "accept_rate": 36,
                "profile_image": "https://www.gravatar.com/avatar/470b0e4f2444335ae1a5f4927a558148?s=128&d=identicon&r=PG&f=1",
                "display_name": "cgao",
                "link": "https://stackoverflow.com/users/2705212/cgao"
            },
            "is_answered": false,
            "view_count": 19,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185860,
            "creation_date": 1524185860,
            "question_id": 49932310,
            "body_markdown": "The [ml.classification.LogisticRegression][1] document points to the [logistic regression in spark.mllib][2], which says:\r\n&gt;We implemented two algorithms to solve logistic regression: mini-batch gradient descent and L-BFGS. We recommend L-BFGS over mini-batch gradient descent for faster convergence.\r\n\r\nSo in spark.mllib there are [LogisticRegressionWithLBFGS][3] and [LogisticRegressionWithSGD][4]. Which one is used in ml.classification.LogisticRegression?\r\n\r\n\r\n  [1]: https://spark.apache.org/docs/2.3.0/ml-classification-regression.html#binomial-logistic-regression\r\n  [2]: https://spark.apache.org/docs/2.3.0/mllib-linear-methods.html#logistic-regression\r\n  [3]: https://spark.apache.org/docs/2.3.0/api/scala/index.html#org.apache.spark.mllib.classification.LogisticRegressionWithLBFGS\r\n  [4]: https://spark.apache.org/docs/2.3.0/api/scala/index.html#org.apache.spark.mllib.classification.LogisticRegressionWithSGD",
            "link": "https://stackoverflow.com/questions/49932310/what-algorithm-does-ml-classification-logisticregression-use-in-spark",
            "title": "What algorithm does ml.classification.LogisticRegression use in spark?",
            "body": "<p>The <a href=\"https://spark.apache.org/docs/2.3.0/ml-classification-regression.html#binomial-logistic-regression\" rel=\"nofollow noreferrer\">ml.classification.LogisticRegression</a> document points to the <a href=\"https://spark.apache.org/docs/2.3.0/mllib-linear-methods.html#logistic-regression\" rel=\"nofollow noreferrer\">logistic regression in spark.mllib</a>, which says:</p>\n\n<blockquote>\n  <p>We implemented two algorithms to solve logistic regression: mini-batch gradient descent and L-BFGS. We recommend L-BFGS over mini-batch gradient descent for faster convergence.</p>\n</blockquote>\n\n<p>So in spark.mllib there are <a href=\"https://spark.apache.org/docs/2.3.0/api/scala/index.html#org.apache.spark.mllib.classification.LogisticRegressionWithLBFGS\" rel=\"nofollow noreferrer\">LogisticRegressionWithLBFGS</a> and <a href=\"https://spark.apache.org/docs/2.3.0/api/scala/index.html#org.apache.spark.mllib.classification.LogisticRegressionWithSGD\" rel=\"nofollow noreferrer\">LogisticRegressionWithSGD</a>. Which one is used in ml.classification.LogisticRegression?</p>\n"
        },
        {
            "tags": [
                "php",
                "laravel-5.6"
            ],
            "owner": {
                "reputation": 205,
                "user_id": 8517929,
                "user_type": "registered",
                "accept_rate": 75,
                "profile_image": "https://www.gravatar.com/avatar/c298955f3de422f89eda88ed5968e91b?s=128&d=identicon&r=PG&f=1",
                "display_name": "Stupid Kid",
                "link": "https://stackoverflow.com/users/8517929/stupid-kid"
            },
            "is_answered": false,
            "view_count": 11,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185853,
            "creation_date": 1524185853,
            "question_id": 49932309,
            "body_markdown": "I&#39;ve added to laravel 5.6 auth registration sending mail:\r\n\r\nRegistersUsers.php\r\n\r\n    event(new Registered($user = $this-&gt;create($request-&gt;all())));\r\n    \r\n    Mail::to($user)-&gt;send(new Welcome($user));\r\n    \r\n    $this-&gt;guard()-&gt;login($user);\r\n    \r\n    if (Mail::failures()) {\r\n        dd(Mail::failures()); - This returns to($user-&gt;email)\r\n    }\r\nMy env file for mail info contains same info I used with phpmailer where it works fine.\r\n\r\n    MAIL_DRIVER=smtp\r\n    MAIL_HOST=mail.my-site.com\r\n    MAIL_PORT=465\r\n    MAIL_USERNAME=noreply@my-site.com\r\n    MAIL_PASSWORD=*******\r\n    MAIL_ENCRYPTION=ssl\r\n\r\nMost answers for this are to make sure I am using right driver. I do, and if I change to some other I get some error.\r\n\r\n***Swift_TransportException\r\nProcess could not be started [The system cannot find the path specified. ]***",
            "link": "https://stackoverflow.com/questions/49932309/laravel-mail-failures-returns-tomail",
            "title": "Laravel mail failures returns to(mail)",
            "body": "<p>I've added to laravel 5.6 auth registration sending mail:</p>\n\n<p>RegistersUsers.php</p>\n\n<pre><code>event(new Registered($user = $this-&gt;create($request-&gt;all())));\n\nMail::to($user)-&gt;send(new Welcome($user));\n\n$this-&gt;guard()-&gt;login($user);\n\nif (Mail::failures()) {\n    dd(Mail::failures()); - This returns to($user-&gt;email)\n}\n</code></pre>\n\n<p>My env file for mail info contains same info I used with phpmailer where it works fine.</p>\n\n<pre><code>MAIL_DRIVER=smtp\nMAIL_HOST=mail.my-site.com\nMAIL_PORT=465\nMAIL_USERNAME=noreply@my-site.com\nMAIL_PASSWORD=*******\nMAIL_ENCRYPTION=ssl\n</code></pre>\n\n<p>Most answers for this are to make sure I am using right driver. I do, and if I change to some other I get some error.</p>\n\n<p><strong><em>Swift_TransportException\nProcess could not be started [The system cannot find the path specified. ]</em></strong></p>\n"
        },
        {
            "tags": [
                "python"
            ],
            "owner": {
                "reputation": 8,
                "user_id": 9336392,
                "user_type": "registered",
                "profile_image": "https://graph.facebook.com/1088466267960657/picture?type=large",
                "display_name": "Ankush Kar",
                "link": "https://stackoverflow.com/users/9336392/ankush-kar"
            },
            "is_answered": true,
            "view_count": 14,
            "closed_date": 1524186973,
            "accepted_answer_id": 49932279,
            "answer_count": 1,
            "score": -2,
            "last_activity_date": 1524185846,
            "creation_date": 1524185311,
            "last_edit_date": 1524185452,
            "question_id": 49932254,
            "body_markdown": "Python program that will take two numbers: a and b.Return a number that is\r\na roundup version of a so that it is multiple of b:\r\nExamples: \r\n\r\n    a=13, b = 5, return 15.\r\n    a=15, b = 3, return 15\r\n    a = 16, b = 3, returns 18&#39;",
            "link": "https://stackoverflow.com/questions/49932254/return-a-number-that-is-a-roundup",
            "closed_reason": "too broad",
            "title": "Return a number that is a roundup",
            "body": "<p>Python program that will take two numbers: a and b.Return a number that is\na roundup version of a so that it is multiple of b:\nExamples: </p>\n\n<pre><code>a=13, b = 5, return 15.\na=15, b = 3, return 15\na = 16, b = 3, returns 18'\n</code></pre>\n"
        },
        {
            "tags": [
                "azure",
                "azure-functions",
                "azure-storage-files"
            ],
            "owner": {
                "reputation": 42,
                "user_id": 1747835,
                "user_type": "registered",
                "accept_rate": 100,
                "profile_image": "https://www.gravatar.com/avatar/d5e08cfc328978c12673eda0251d9a09?s=128&d=identicon&r=PG",
                "display_name": "Justin Neff",
                "link": "https://stackoverflow.com/users/1747835/justin-neff"
            },
            "is_answered": true,
            "view_count": 88,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185825,
            "creation_date": 1523550330,
            "question_id": 49801605,
            "body_markdown": "When creating an Azure function with an Dedicated (Standard) App Service Plan  the file service I&#39;m expecting to get &quot;linked&quot; up isn&#39;t getting &quot;linked&quot; to the storage account. However the Storage account does get created correctly. When I go to the Azure Storage Accounts blade and find the file storage, Azure doesn&#39;t have the file service linked to the Storage Account. I don&#39;t see any linked File Shares when using the Windows desktop software, Microsoft Azure Storage Explorer (0.9.6).\r\n\r\nWhen I go to the Azure Function&#39;s Advanced Tools (Kudo) I can see the storage account folders &quot;Data&quot;, &quot;LogFiles&quot;, and &quot;Site&quot; with the wwwroot I&#39;m expecting to find. However due to certain network restrictions I can&#39;t upload the code through the web-site so that option is out.\r\n\r\nWhen creating a Consumption based plan everything links up nicely and I can manage them in the Azure Storage Explorer app. How can I get my already created File Service File Share linked to my already created Azure Function so I can manage it in the desktop app and get things appearing correctly linked up in the Storage Explorer?",
            "link": "https://stackoverflow.com/questions/49801605/azure-function-storage-account-not-linking-file-service-on-create",
            "title": "Azure Function Storage Account Not Linking File Service On Create",
            "body": "<p>When creating an Azure function with an Dedicated (Standard) App Service Plan  the file service I'm expecting to get \"linked\" up isn't getting \"linked\" to the storage account. However the Storage account does get created correctly. When I go to the Azure Storage Accounts blade and find the file storage, Azure doesn't have the file service linked to the Storage Account. I don't see any linked File Shares when using the Windows desktop software, Microsoft Azure Storage Explorer (0.9.6).</p>\n\n<p>When I go to the Azure Function's Advanced Tools (Kudo) I can see the storage account folders \"Data\", \"LogFiles\", and \"Site\" with the wwwroot I'm expecting to find. However due to certain network restrictions I can't upload the code through the web-site so that option is out.</p>\n\n<p>When creating a Consumption based plan everything links up nicely and I can manage them in the Azure Storage Explorer app. How can I get my already created File Service File Share linked to my already created Azure Function so I can manage it in the desktop app and get things appearing correctly linked up in the Storage Explorer?</p>\n"
        },
        {
            "tags": [
                "docker",
                "docker-windows"
            ],
            "owner": {
                "reputation": 618,
                "user_id": 601182,
                "user_type": "registered",
                "accept_rate": 62,
                "profile_image": "https://www.gravatar.com/avatar/64ebc54e23e49cf1663092f3944bc835?s=128&d=identicon&r=PG",
                "display_name": "Aftab Naveed",
                "link": "https://stackoverflow.com/users/601182/aftab-naveed"
            },
            "is_answered": false,
            "view_count": 9,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185825,
            "creation_date": 1524143095,
            "question_id": 49921887,
            "body_markdown": "I have an application running inside docker which programmatically creates files. Most of the time files are created properly and then deleted. Sometimes a file with unknown permission is created\r\n\r\n    drwxrwxrwx 0 root root 4096 Apr 17 21:41 ..\r\n    -????????? ? ?    ?       ?            ? __CG__NarwhalLayoutDomainEntitiesTheme.php\r\n    -rwxrwxrwx 1 root root 8314 Apr 19 12:46 __CG__NarwhalLayoutDomainEntitiesTheme.php.5ad802fc34f6b0.60147712\r\n\r\nWhen that happens though I can see the file in Windows Explorer when I try to delete it I get this error:\r\n\r\n    &quot;You need permission to perform this action&quot;\r\n    &quot;You require permission from the computer&#39;s administrator to make changes to this file&quot;\r\n\r\nI even tried as administrator but it still won&#39;t work and when I try to delete the file from inside the container shell. I get this error:\r\n\r\nrm: cannot remove &#39;__CG__NarwhalLayoutDomainEntitiesTheme.php&#39;: No such file or directory\r\n\r\nThis problem goes away when I restart the docker but I don&#39;t want to restart docker every few minutes.",
            "link": "https://stackoverflow.com/questions/49921887/docker-unknown-file-permissions-windows-10",
            "title": "Docker Unknown File Permissions Windows 10",
            "body": "<p>I have an application running inside docker which programmatically creates files. Most of the time files are created properly and then deleted. Sometimes a file with unknown permission is created</p>\n\n<pre><code>drwxrwxrwx 0 root root 4096 Apr 17 21:41 ..\n-????????? ? ?    ?       ?            ? __CG__NarwhalLayoutDomainEntitiesTheme.php\n-rwxrwxrwx 1 root root 8314 Apr 19 12:46 __CG__NarwhalLayoutDomainEntitiesTheme.php.5ad802fc34f6b0.60147712\n</code></pre>\n\n<p>When that happens though I can see the file in Windows Explorer when I try to delete it I get this error:</p>\n\n<pre><code>\"You need permission to perform this action\"\n\"You require permission from the computer's administrator to make changes to this file\"\n</code></pre>\n\n<p>I even tried as administrator but it still won't work and when I try to delete the file from inside the container shell. I get this error:</p>\n\n<p>rm: cannot remove '__CG__NarwhalLayoutDomainEntitiesTheme.php': No such file or directory</p>\n\n<p>This problem goes away when I restart the docker but I don't want to restart docker every few minutes.</p>\n"
        },
        {
            "tags": [
                "python",
                "csv",
                "beautifulsoup"
            ],
            "owner": {
                "reputation": 5,
                "user_id": 9661706,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-XdUIqdMkCWA/AAAAAAAAAAI/AAAAAAAAAAA/4252rscbv5M/photo.jpg?sz=128",
                "display_name": "Nate Walker",
                "link": "https://stackoverflow.com/users/9661706/nate-walker"
            },
            "is_answered": true,
            "view_count": 32,
            "accepted_answer_id": 49932284,
            "answer_count": 2,
            "score": 0,
            "last_activity_date": 1524185824,
            "creation_date": 1524185000,
            "question_id": 49932226,
            "body_markdown": "In the code below, I have successfully scraped a list of every MLB team and their corresponding win probability for the day (April 18th). I would like to export this data to a CSV file, but when I write the code, only one team and win probability gets exported. Does anyone know why this happens? I&#39;m thinking there needs to be another for loop written with the CSV writer but I am not exactly sure how to do that with two separate sources of scraped data (team name and win probability) Thanks in advance!\r\n\r\n    import requests\r\n    import csv\r\n    from bs4 import BeautifulSoup\r\n\r\n    page=requests.get(&#39;https://www.fangraphs.com/livescoreboard.aspx?date=2018- \r\n    04-18&#39;)\r\n    soup=BeautifulSoup(page.text, &#39;html.parser&#39;)\r\n\r\n\r\n    [link.decompose() for link in soup.find_all(class_=&#39;lineup&#39;)]\r\n\r\n    f=csv.writer(open(&#39;Win_Probability.csv&#39;,&#39;w&#39;))\r\n    f.writerow([&#39;Teams&#39;,&#39;Win_Prob&#39;])\r\n\r\n    team_name_list=soup.find(class_=&#39;RadAjaxPanel&#39;)\r\n    team_name_list_items=team_name_list.find_all(&#39;a&#39;)\r\n\r\n\r\n    for team_name in team_name_list_items:\r\n      teams=team_name.contents[0]\r\n      print(teams)\r\n\r\n    winprob_list=soup.find(class_=&#39;RadAjaxPanel&#39;)\r\n    winprob_list_items=winprob_list.find_all(&#39;td&#39;,attrs={&#39;style&#39;:&#39;border:1px \r\n    solid black;&#39;})\r\n\r\n    for winprob in winprob_list_items:\r\n      winprobperc=winprob.contents[0]\r\n      print(winprobperc)\r\n\r\n\r\n    f.writerow([teams,winprobperc])",
            "link": "https://stackoverflow.com/questions/49932226/writing-and-saving-csv-file-from-scraping-data",
            "title": "Writing and Saving CSV File From Scraping Data",
            "body": "<p>In the code below, I have successfully scraped a list of every MLB team and their corresponding win probability for the day (April 18th). I would like to export this data to a CSV file, but when I write the code, only one team and win probability gets exported. Does anyone know why this happens? I'm thinking there needs to be another for loop written with the CSV writer but I am not exactly sure how to do that with two separate sources of scraped data (team name and win probability) Thanks in advance!</p>\n\n<pre><code>import requests\nimport csv\nfrom bs4 import BeautifulSoup\n\npage=requests.get('https://www.fangraphs.com/livescoreboard.aspx?date=2018- \n04-18')\nsoup=BeautifulSoup(page.text, 'html.parser')\n\n\n[link.decompose() for link in soup.find_all(class_='lineup')]\n\nf=csv.writer(open('Win_Probability.csv','w'))\nf.writerow(['Teams','Win_Prob'])\n\nteam_name_list=soup.find(class_='RadAjaxPanel')\nteam_name_list_items=team_name_list.find_all('a')\n\n\nfor team_name in team_name_list_items:\n  teams=team_name.contents[0]\n  print(teams)\n\nwinprob_list=soup.find(class_='RadAjaxPanel')\nwinprob_list_items=winprob_list.find_all('td',attrs={'style':'border:1px \nsolid black;'})\n\nfor winprob in winprob_list_items:\n  winprobperc=winprob.contents[0]\n  print(winprobperc)\n\n\nf.writerow([teams,winprobperc])\n</code></pre>\n"
        },
        {
            "tags": [
                "php",
                "wordpress",
                "woocommerce"
            ],
            "owner": {
                "reputation": 11,
                "user_id": 5832863,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/bc62cd665986866764ce27c1f44b79cf?s=128&d=identicon&r=PG",
                "display_name": "Luke Mitchell",
                "link": "https://stackoverflow.com/users/5832863/luke-mitchell"
            },
            "is_answered": false,
            "view_count": 17,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185783,
            "creation_date": 1524181393,
            "question_id": 49931820,
            "body_markdown": "I have a WooCommerce store which I need the product price to update if the quantity is changed. Example if you wish to buy 3 mugs for $50, then the price will change to $150 on the single product page automatically, and as you increase the quantity the price shown will increase by $50 each time. \r\n\r\nMost of my store is variable products, so the price must update on variable products as well as simple products.\r\n\r\nThe below code is almost there, but still not right. It doesn&#39;t work very well on variable products and doesn&#39;t update the main product regular price. \r\n\r\n    \r\n    add_action( &#39;woocommerce_single_product_summary&#39;, &#39;woocommerce_total_product_price&#39;, 31 );\r\n    function woocommerce_total_product_price() {\r\n        global $woocommerce, $product;\r\n        \r\n\r\n        echo sprintf(&#39;&lt;div id=&quot;product_total_price&quot; style=&quot;margin-bottom:20px;display:none&quot;&gt;%s %s&lt;/div&gt;&#39;,__(&#39;Product Total:&#39;,&#39;woocommerce&#39;),&#39;&lt;span class=&quot;price&quot;&gt;&#39;.$product-&gt;get_price().&#39;&lt;/span&gt;&#39;);\r\n        echo sprintf(&#39;&lt;div id=&quot;cart_total_price&quot; style=&quot;margin-bottom:20px;display:none&quot;&gt;%s %s&lt;/div&gt;&#39;,__(&#39;Cart Total:&#39;,&#39;woocommerce&#39;),&#39;&lt;span class=&quot;price&quot;&gt;&#39;.$product-&gt;get_price().&#39;&lt;/span&gt;&#39;);\r\n        ?&gt;\r\n            &lt;script&gt;\r\n                jQuery(function($){\r\n                    var price = &lt;?php echo $product-&gt;get_price(); ?&gt;,\r\n                        current_cart_total = &lt;?php echo $woocommerce-&gt;cart-&gt;cart_contents_total; ?&gt;,\r\n                        currency = &#39;&lt;?php echo get_woocommerce_currency_symbol(); ?&gt;&#39;;\r\n     \r\n                    $(&#39;[name=quantity]&#39;).change(function(){\r\n                        if (!(this.value &lt; 1)) {\r\n                            var product_total = parseFloat(price * this.value),\r\n                            cart_total = parseFloat(product_total + current_cart_total);\r\n     \r\n                            $(&#39;#product_total_price .price&#39;).html( currency + product_total.toFixed(2));\r\n                            $(&#39;#cart_total_price .price&#39;).html( currency + cart_total.toFixed(2));\r\n                        }\r\n                        $(&#39;#product_total_price,#cart_total_price&#39;).toggle(!(this.value &lt;= 1));\r\n     \r\n                    });\r\n                });\r\n            &lt;/script&gt;\r\n        &lt;?php\r\n    }\r\n\r\nAny help would be appreciated! ",
            "link": "https://stackoverflow.com/questions/49931820/woocommerce-update-price-of-a-single-product-when-the-quantity-is-changed",
            "title": "WooCommerce update price of a single product when the quantity is changed",
            "body": "<p>I have a WooCommerce store which I need the product price to update if the quantity is changed. Example if you wish to buy 3 mugs for $50, then the price will change to $150 on the single product page automatically, and as you increase the quantity the price shown will increase by $50 each time. </p>\n\n<p>Most of my store is variable products, so the price must update on variable products as well as simple products.</p>\n\n<p>The below code is almost there, but still not right. It doesn't work very well on variable products and doesn't update the main product regular price. </p>\n\n<pre><code>add_action( 'woocommerce_single_product_summary', 'woocommerce_total_product_price', 31 );\nfunction woocommerce_total_product_price() {\n    global $woocommerce, $product;\n\n\n    echo sprintf('&lt;div id=\"product_total_price\" style=\"margin-bottom:20px;display:none\"&gt;%s %s&lt;/div&gt;',__('Product Total:','woocommerce'),'&lt;span class=\"price\"&gt;'.$product-&gt;get_price().'&lt;/span&gt;');\n    echo sprintf('&lt;div id=\"cart_total_price\" style=\"margin-bottom:20px;display:none\"&gt;%s %s&lt;/div&gt;',__('Cart Total:','woocommerce'),'&lt;span class=\"price\"&gt;'.$product-&gt;get_price().'&lt;/span&gt;');\n    ?&gt;\n        &lt;script&gt;\n            jQuery(function($){\n                var price = &lt;?php echo $product-&gt;get_price(); ?&gt;,\n                    current_cart_total = &lt;?php echo $woocommerce-&gt;cart-&gt;cart_contents_total; ?&gt;,\n                    currency = '&lt;?php echo get_woocommerce_currency_symbol(); ?&gt;';\n\n                $('[name=quantity]').change(function(){\n                    if (!(this.value &lt; 1)) {\n                        var product_total = parseFloat(price * this.value),\n                        cart_total = parseFloat(product_total + current_cart_total);\n\n                        $('#product_total_price .price').html( currency + product_total.toFixed(2));\n                        $('#cart_total_price .price').html( currency + cart_total.toFixed(2));\n                    }\n                    $('#product_total_price,#cart_total_price').toggle(!(this.value &lt;= 1));\n\n                });\n            });\n        &lt;/script&gt;\n    &lt;?php\n}\n</code></pre>\n\n<p>Any help would be appreciated! </p>\n"
        },
        {
            "tags": [
                "android",
                "automated-tests",
                "appium"
            ],
            "owner": {
                "reputation": 385,
                "user_id": 6521664,
                "user_type": "registered",
                "accept_rate": 50,
                "profile_image": "https://graph.facebook.com/10209062777223388/picture?type=large",
                "display_name": "Chattip Soontaku",
                "link": "https://stackoverflow.com/users/6521664/chattip-soontaku"
            },
            "is_answered": true,
            "view_count": 11146,
            "protected_date": 1479303691,
            "accepted_answer_id": 40168992,
            "answer_count": 2,
            "score": 14,
            "last_activity_date": 1524185783,
            "creation_date": 1476875452,
            "last_edit_date": 1476875592,
            "question_id": 40129794,
            "body_markdown": "I installed appium-1.5.3.dmg But when I click the stethoscope button the Appium is show error :\r\n\r\n     ✔ Xcode is installed at: /Library/Developer/CommandLineTools\r\n    [Error: Could not detect Mac OS X Version from sw_vers output: &#39;10.12\r\n    &#39;]\r\n\r\n![enter image description here][1]\r\n\r\n\r\n  [1]: https://i.stack.imgur.com/fSdYv.png\r\n\r\nThank you!",
            "link": "https://stackoverflow.com/questions/40129794/how-to-fix-error-could-not-detect-mac-os-x-version-from-sw-vers-output-10-12",
            "title": "How to fix error &quot;Could not detect Mac OS X Version from sw_vers output: &#39;10.12 &#39;&quot; from Appium",
            "body": "<p>I installed appium-1.5.3.dmg But when I click the stethoscope button the Appium is show error :</p>\n\n<pre><code> ✔ Xcode is installed at: /Library/Developer/CommandLineTools\n[Error: Could not detect Mac OS X Version from sw_vers output: '10.12\n']\n</code></pre>\n\n<p><img src=\"https://i.stack.imgur.com/fSdYv.png\" alt=\"enter image description here\"></p>\n\n<p>Thank you!</p>\n"
        },
        {
            "tags": [
                "jasper-reports"
            ],
            "owner": {
                "reputation": 3,
                "user_id": 8036435,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/b2d205d16c534cb6ce672c8a67ceadc3?s=128&d=identicon&r=PG",
                "display_name": "Kelvin Santiago",
                "link": "https://stackoverflow.com/users/8036435/kelvin-santiago"
            },
            "is_answered": false,
            "view_count": 200,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185771,
            "creation_date": 1508290661,
            "last_edit_date": 1508313313,
            "question_id": 46801563,
            "body_markdown": "I&#39;m passing the object &quot;OrdemServico&quot; as parameter to report.\r\n\r\n    OrdemServico ordemServico = new OrdemServico();  \r\n    ordemServico.setCodigo(&quot;CA123&quot;);\r\n    mapa.put(&quot;ORDEMSERVICO&quot;, ordemServico);\r\n\r\nAnd on the .jrxml i added a parameter with same name of the parameter in map. how i can get the fields of the object? Tnks",
            "link": "https://stackoverflow.com/questions/46801563/how-get-fields-of-object-in-parameter-ireport",
            "title": "How get fields of object in parameter iReport",
            "body": "<p>I'm passing the object \"OrdemServico\" as parameter to report.</p>\n\n<pre><code>OrdemServico ordemServico = new OrdemServico();  \nordemServico.setCodigo(\"CA123\");\nmapa.put(\"ORDEMSERVICO\", ordemServico);\n</code></pre>\n\n<p>And on the .jrxml i added a parameter with same name of the parameter in map. how i can get the fields of the object? Tnks</p>\n"
        },
        {
            "tags": [
                "c#",
                "aspnetboilerplate"
            ],
            "owner": {
                "reputation": 33,
                "user_id": 8123296,
                "user_type": "registered",
                "accept_rate": 83,
                "profile_image": "https://www.gravatar.com/avatar/74162e6d80b199b6478ba1dc7c3b6fc2?s=128&d=identicon&r=PG&f=1",
                "display_name": "Mohammad Shadmehr",
                "link": "https://stackoverflow.com/users/8123296/mohammad-shadmehr"
            },
            "is_answered": false,
            "view_count": 69,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185745,
            "creation_date": 1524029142,
            "last_edit_date": 1524185745,
            "question_id": 49891772,
            "body_markdown": "I am looking for a way to copy the value of some properties of a class to another table in DB. Is there any particular way that I can mark properties of a class as one which their values need to be stored in 2 places (one as part of its own domain object and the other in another domain object, say Summary)?\r\n\r\nI a having a Audited entity as below:\r\n\r\n    public class Audited \r\n    {\r\n        public virtual int Id{ get; set; }\r\n\r\n        public virtual string FieldName{ get; set; }\r\n\r\n        public virtual string FieldValue{ get; set; }\r\n    }\r\n\r\nAnd other entities such as:\r\n\r\n    public class Plan : FullAuditedEntity\r\n    {\r\n        [ToBeAudited]\r\n        public virtual int PlanName{ get; set; }\r\n\r\n        public DateTime Date { get; set; }\r\n    }\r\n\r\nI am looking for a way to be able to mark properties of classes with an attribute (eg. [ToBeAudited]) so that property value gets copied into the Audited table in on insert or update\r\n\r\nI have added the [Audited] (`Abp.Auditing`) to `PlanName` and getting the following error in log:\r\n\r\n&gt; ERROR 2018-04-20 10:02:09,286 [5    ] Mvc.ExceptionHandling.AbpExceptionFilter - Object reference not set to an instance of an object.\r\nSystem.NullReferenceException: Object reference not set to an instance of an object.\r\n   at Abp.EntityHistory.EntityHistoryHelper.ShouldSavePropertyHistory(PropertyEntry propertyEntry, Boolean defaultValue)\r\n   at Abp.EntityHistory.EntityHistoryHelper.GetPropertyChanges(EntityEntry entityEntry)\r\n   at Abp.EntityHistory.EntityHistoryHelper.CreateEntityChangeInfo(EntityEntry entityEntry)\r\n   at Abp.EntityHistory.EntityHistoryHelper.CreateEntityChangeSet(ICollection`1 entityEntries)\r\n   at Abp.Zero.EntityFrameworkCore.AbpZeroCommonDbContext`3.&lt;SaveChangesAsync&gt;d__98.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Abp.EntityFrameworkCore.Uow.EfCoreUnitOfWork.&lt;SaveChangesInDbContextAsync&gt;d__20.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Abp.EntityFrameworkCore.Uow.EfCoreUnitOfWork.&lt;SaveChangesAsync&gt;d__12.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Abp.EntityFrameworkCore.Uow.EfCoreUnitOfWork.&lt;CompleteUowAsync&gt;d__14.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Abp.Domain.Uow.UnitOfWorkBase.&lt;CompleteAsync&gt;d__57.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Abp.AspNetCore.Mvc.Uow.AbpUowActionFilter.&lt;OnActionExecutionAsync&gt;d__4.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.&lt;InvokeNextActionFilterAsync&gt;d__10.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)\r\n   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State&amp; next, Scope&amp; scope, Object&amp; state, Boolean&amp; isCompleted)\r\n   at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.&lt;InvokeInnerFilterAsync&gt;d__14.MoveNext()\r\n--- End of stack trace from previous location where exception was thrown ---\r\n   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\r\n   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\r\n   at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.&lt;InvokeNextExceptionFilterAsync&gt;d__23.MoveNext()\r\n\r\n",
            "link": "https://stackoverflow.com/questions/49891772/storing-a-property-value-into-another-table",
            "title": "Storing a property value into another table",
            "body": "<p>I am looking for a way to copy the value of some properties of a class to another table in DB. Is there any particular way that I can mark properties of a class as one which their values need to be stored in 2 places (one as part of its own domain object and the other in another domain object, say Summary)?</p>\n\n<p>I a having a Audited entity as below:</p>\n\n<pre><code>public class Audited \n{\n    public virtual int Id{ get; set; }\n\n    public virtual string FieldName{ get; set; }\n\n    public virtual string FieldValue{ get; set; }\n}\n</code></pre>\n\n<p>And other entities such as:</p>\n\n<pre><code>public class Plan : FullAuditedEntity\n{\n    [ToBeAudited]\n    public virtual int PlanName{ get; set; }\n\n    public DateTime Date { get; set; }\n}\n</code></pre>\n\n<p>I am looking for a way to be able to mark properties of classes with an attribute (eg. [ToBeAudited]) so that property value gets copied into the Audited table in on insert or update</p>\n\n<p>I have added the [Audited] (<code>Abp.Auditing</code>) to <code>PlanName</code> and getting the following error in log:</p>\n\n<blockquote>\n  <p>ERROR 2018-04-20 10:02:09,286 [5    ] Mvc.ExceptionHandling.AbpExceptionFilter - Object reference not set to an instance of an object.\n  System.NullReferenceException: Object reference not set to an instance of an object.\n     at Abp.EntityHistory.EntityHistoryHelper.ShouldSavePropertyHistory(PropertyEntry propertyEntry, Boolean defaultValue)\n     at Abp.EntityHistory.EntityHistoryHelper.GetPropertyChanges(EntityEntry entityEntry)\n     at Abp.EntityHistory.EntityHistoryHelper.CreateEntityChangeInfo(EntityEntry entityEntry)\n     at Abp.EntityHistory.EntityHistoryHelper.CreateEntityChangeSet(ICollection<code>1 entityEntries)\n     at Abp.Zero.EntityFrameworkCore.AbpZeroCommonDbContext</code>3.d__98.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Abp.EntityFrameworkCore.Uow.EfCoreUnitOfWork.d__20.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Abp.EntityFrameworkCore.Uow.EfCoreUnitOfWork.d__12.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Abp.EntityFrameworkCore.Uow.EfCoreUnitOfWork.d__14.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Abp.Domain.Uow.UnitOfWorkBase.d__57.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Abp.AspNetCore.Mvc.Uow.AbpUowActionFilter.d__4.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.d__10.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Rethrow(ActionExecutedContext context)\n     at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.Next(State&amp; next, Scope&amp; scope, Object&amp; state, Boolean&amp; isCompleted)\n     at Microsoft.AspNetCore.Mvc.Internal.ControllerActionInvoker.d__14.MoveNext()\n  --- End of stack trace from previous location where exception was thrown ---\n     at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)\n     at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)\n     at Microsoft.AspNetCore.Mvc.Internal.ResourceInvoker.d__23.MoveNext()</p>\n</blockquote>\n"
        },
        {
            "tags": [
                "javascript",
                "node.js",
                "download",
                "unzip"
            ],
            "owner": {
                "reputation": 14,
                "user_id": 4950937,
                "user_type": "registered",
                "accept_rate": 50,
                "profile_image": "https://www.gravatar.com/avatar/18ac292b86cee2e14696374b04c1a8d1?s=128&d=identicon&r=PG&f=1",
                "display_name": "akshita007",
                "link": "https://stackoverflow.com/users/4950937/akshita007"
            },
            "is_answered": false,
            "view_count": 20,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185722,
            "creation_date": 1524185722,
            "question_id": 49932297,
            "body_markdown": "I need to write two separate components - one is a downloader which takes a file path(or a FILE DESCRIPTOR whichever seems better) and url and downloads whatever file is present in that url into the file path. The code that I have written so far for the downloader component is :  https://gist.github.com/akshita31/b3df6f8b45da342b7a8a4bd24cdf6c1c\r\n\r\nOnce the download is complete, I have an installer component that unzips the file present at that file descriptor and puts them at a destination path. Code for the installer is: https://gist.github.com/akshita31/9e86c1090622c462cc357b3374acba4c\r\n\r\nI have a third utility that creates a tmpFile(using npm tmp) - https://gist.github.com/akshita31/92878afbb0943d3a95e14193ddf9f64b.\r\n\r\nFinally I do:\r\n\r\n    tmpFile = await createTmpFile();\r\n    await DownloadFile(tmpFile.fd, pkg.description, pkg.url, pkg.fallbackUrl, eventStream, provider);\r\n    await InstallPackage(tmpFile.fd, pkg.description, pkg.installPath, pkg.binaries, eventStream);\r\n\r\nBut in the installation I sometimes get - &#39;End of central directory record signature not found&#39; which to me suggests that the file was not completely downloaded and put in to the desired path and we are trying to read it before that. After a lot of googling I somehow understood that it might have to do something with &#39;finish&#39; event on the WriteStream, but if I resolve the downloader just on that my program waits a lot of time and then times out. What is it that I am doing wrong ?",
            "link": "https://stackoverflow.com/questions/49932297/downloading-a-zip-into-a-file-and-unzipping-it-using-nodejs-how-to-detect-when",
            "title": "Downloading a zip into a file and unzipping it using nodejs - how to detect when the download is complete and it is safe to unzip",
            "body": "<p>I need to write two separate components - one is a downloader which takes a file path(or a FILE DESCRIPTOR whichever seems better) and url and downloads whatever file is present in that url into the file path. The code that I have written so far for the downloader component is :  <a href=\"https://gist.github.com/akshita31/b3df6f8b45da342b7a8a4bd24cdf6c1c\" rel=\"nofollow noreferrer\">https://gist.github.com/akshita31/b3df6f8b45da342b7a8a4bd24cdf6c1c</a></p>\n\n<p>Once the download is complete, I have an installer component that unzips the file present at that file descriptor and puts them at a destination path. Code for the installer is: <a href=\"https://gist.github.com/akshita31/9e86c1090622c462cc357b3374acba4c\" rel=\"nofollow noreferrer\">https://gist.github.com/akshita31/9e86c1090622c462cc357b3374acba4c</a></p>\n\n<p>I have a third utility that creates a tmpFile(using npm tmp) - <a href=\"https://gist.github.com/akshita31/92878afbb0943d3a95e14193ddf9f64b\" rel=\"nofollow noreferrer\">https://gist.github.com/akshita31/92878afbb0943d3a95e14193ddf9f64b</a>.</p>\n\n<p>Finally I do:</p>\n\n<pre><code>tmpFile = await createTmpFile();\nawait DownloadFile(tmpFile.fd, pkg.description, pkg.url, pkg.fallbackUrl, eventStream, provider);\nawait InstallPackage(tmpFile.fd, pkg.description, pkg.installPath, pkg.binaries, eventStream);\n</code></pre>\n\n<p>But in the installation I sometimes get - 'End of central directory record signature not found' which to me suggests that the file was not completely downloaded and put in to the desired path and we are trying to read it before that. After a lot of googling I somehow understood that it might have to do something with 'finish' event on the WriteStream, but if I resolve the downloader just on that my program waits a lot of time and then times out. What is it that I am doing wrong ?</p>\n"
        },
        {
            "tags": [
                "php",
                "jquery",
                "mysql",
                "codeigniter",
                "grocery-crud"
            ],
            "owner": {
                "reputation": 39,
                "user_id": 3709356,
                "user_type": "registered",
                "accept_rate": 73,
                "profile_image": "https://i.stack.imgur.com/G9XId.jpg?s=128&g=1",
                "display_name": "ryuusoultaker",
                "link": "https://stackoverflow.com/users/3709356/ryuusoultaker"
            },
            "is_answered": true,
            "view_count": 742,
            "accepted_answer_id": 35608276,
            "answer_count": 2,
            "score": 1,
            "last_activity_date": 1524185716,
            "creation_date": 1424739401,
            "question_id": 28686495,
            "body_markdown": "I face this error when I&#39;m trying to save the add form from Grocery Crud..\r\n\r\n![enter image description here][1]\r\n\r\n\r\n  [1]: http://i.stack.imgur.com/hhXLe.png\r\n\r\nthis error only show up at my server where i change the http port into 88 for port forwarding issue, but the script were working normally in my localhost at my own laptop..\r\n \r\ndid the port change impact the grocery crud? can some one tell me how to fix it since I really have to change the port since port 80 and 8080 is already used..",
            "link": "https://stackoverflow.com/questions/28686495/an-error-occurred-on-insert-in-grocerycrud",
            "title": "An error occurred on insert in GroceryCrud",
            "body": "<p>I face this error when I'm trying to save the add form from Grocery Crud..</p>\n\n<p><img src=\"https://i.stack.imgur.com/hhXLe.png\" alt=\"enter image description here\"></p>\n\n<p>this error only show up at my server where i change the http port into 88 for port forwarding issue, but the script were working normally in my localhost at my own laptop..</p>\n\n<p>did the port change impact the grocery crud? can some one tell me how to fix it since I really have to change the port since port 80 and 8080 is already used..</p>\n"
        },
        {
            "tags": [
                "disassembly",
                "powerpc"
            ],
            "owner": {
                "reputation": 8,
                "user_id": 7713525,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-dDm6uzLZMno/AAAAAAAAAAI/AAAAAAAAABk/UKQQ85HkrXU/photo.jpg?sz=128",
                "display_name": "xiaokang yin",
                "link": "https://stackoverflow.com/users/7713525/xiaokang-yin"
            },
            "is_answered": false,
            "view_count": 27,
            "answer_count": 2,
            "score": 0,
            "last_activity_date": 1524185684,
            "creation_date": 1524064333,
            "last_edit_date": 1524151124,
            "question_id": 49903280,
            "body_markdown": "Given a PowerPC binary file(ELF), we can disassembly it, but how to recoginze the functions as the IDA pro does? Is there an algorithm?",
            "link": "https://stackoverflow.com/questions/49903280/how-to-recognize-functions-start-address-and-end-address-in-binary-of-powerpc",
            "title": "How to recognize functions start address and end address in binary of PowerPC?",
            "body": "<p>Given a PowerPC binary file(ELF), we can disassembly it, but how to recoginze the functions as the IDA pro does? Is there an algorithm?</p>\n"
        },
        {
            "tags": [
                "c#",
                ".net",
                "asp.net-web-api",
                "async-await",
                "asp.net-web-api2"
            ],
            "owner": {
                "reputation": 123,
                "user_id": 9393635,
                "user_type": "registered",
                "accept_rate": 33,
                "profile_image": "https://www.gravatar.com/avatar/2a956540dcbaa39914322d6a5e442125?s=128&d=identicon&r=PG&f=1",
                "display_name": "user9393635",
                "link": "https://stackoverflow.com/users/9393635/user9393635"
            },
            "is_answered": true,
            "view_count": 51,
            "answer_count": 1,
            "score": 1,
            "last_activity_date": 1524185681,
            "creation_date": 1524183125,
            "question_id": 49932014,
            "body_markdown": "I came across the following comment on SO:\r\n\r\n&quot;Also use of the async await pattern with newer versions of .Net will give the efficiencies closer to that of the event-driven architectures like NodeJS and Nginx&quot;\r\n\r\nI inherited a Web API app where basically all methods in all layers leverage the async/await pattern.  The comment above about the performance of async/await is pretty compelling.  Is it proper to implement async/await on all methods of all layers in a Web API app? Are there any scenarios where I should avoid await/async or be careful to not overuse it?",
            "link": "https://stackoverflow.com/questions/49932014/how-extensively-should-the-async-await-pattern-be-used",
            "title": "how extensively should the async/await pattern be used?",
            "body": "<p>I came across the following comment on SO:</p>\n\n<p>\"Also use of the async await pattern with newer versions of .Net will give the efficiencies closer to that of the event-driven architectures like NodeJS and Nginx\"</p>\n\n<p>I inherited a Web API app where basically all methods in all layers leverage the async/await pattern.  The comment above about the performance of async/await is pretty compelling.  Is it proper to implement async/await on all methods of all layers in a Web API app? Are there any scenarios where I should avoid await/async or be careful to not overuse it?</p>\n"
        },
        {
            "tags": [
                "python",
                "string",
                "split"
            ],
            "owner": {
                "reputation": 89,
                "user_id": 3874743,
                "user_type": "registered",
                "accept_rate": 44,
                "profile_image": "https://graph.facebook.com/1374426658/picture?type=large",
                "display_name": "Federico Cuozzo",
                "link": "https://stackoverflow.com/users/3874743/federico-cuozzo"
            },
            "is_answered": true,
            "view_count": 138,
            "answer_count": 2,
            "score": 0,
            "last_activity_date": 1524185675,
            "creation_date": 1504278518,
            "question_id": 46003194,
            "body_markdown": "I&#39;d want to remove stopwords from a sentence.\r\nI&#39;ve this piece of code:\r\n\r\n    splitted = text.split()\r\n\r\n    for index, word in enumerate(splitted):\r\n        if word in self.stopWords:\r\n            del splitted[index]\r\n\r\n    text = &quot; &quot;.join(splitted)\r\n\r\nstopWords is updated with this instruction `self.stopWords.update([&#39;.&#39;, &#39;,&#39;, &quot;\\&quot;&quot;, &quot;\\&#39;&quot;, &#39;?&#39;, &#39;!&#39;, &#39;:&#39;, &#39;;&#39;, &#39;(&#39;, &#39;)&#39;, &#39;[&#39;, &#39;]&#39;, &#39;{&#39;, &#39;}&#39;, &#39;),&#39;, &#39;],&#39;, &#39;},&#39;, &#39;&quot;,&#39;, &quot;&#39;,&quot;, &#39;&quot;)&#39;, &#39;&quot;]&#39;, &#39;&quot;}&#39;, &quot;-&quot;, &quot;--&quot;, &#39;\\&quot;.&#39;, &quot;\\&#39;.&quot;, &#39;/&#39;, &#39;).&#39;, &#39;-&#39;, &#39;--&#39;, &#39;%&#39;, &#39;&#176;\\&#39;&#39;, &#39;(-&#39;, &#39;(&quot;&#39;, &#39;.&quot;&#39;, &#39;.),&#39;, &#39;);&#39;, &#39;–&#39;, &#39;$&#39;, &#39;a&#39;])` but, for example, the letter &#39;a&#39;, such as &#39;.&#39;, or &#39;;&#39; isn&#39;t deleted from the sentence.\r\n\r\nWhat would I do?",
            "link": "https://stackoverflow.com/questions/46003194/remove-stopwords-from-sentences",
            "title": "Remove stopwords from sentences",
            "body": "<p>I'd want to remove stopwords from a sentence.\nI've this piece of code:</p>\n\n<pre><code>splitted = text.split()\n\nfor index, word in enumerate(splitted):\n    if word in self.stopWords:\n        del splitted[index]\n\ntext = \" \".join(splitted)\n</code></pre>\n\n<p>stopWords is updated with this instruction <code>self.stopWords.update(['.', ',', \"\\\"\", \"\\'\", '?', '!', ':', ';', '(', ')', '[', ']', '{', '}', '),', '],', '},', '\",', \"',\", '\")', '\"]', '\"}', \"-\", \"--\", '\\\".', \"\\'.\", '/', ').', '-', '--', '%', '°\\'', '(-', '(\"', '.\"', '.),', ');', '–', '$', 'a'])</code> but, for example, the letter 'a', such as '.', or ';' isn't deleted from the sentence.</p>\n\n<p>What would I do?</p>\n"
        },
        {
            "tags": [
                "excel",
                "excel-vba"
            ],
            "owner": {
                "reputation": 8,
                "user_id": 5665723,
                "user_type": "registered",
                "profile_image": "https://www.gravatar.com/avatar/bbae50ba6832468eaed47dd16cf4656c?s=128&d=identicon&r=PG&f=1",
                "display_name": "KWusf",
                "link": "https://stackoverflow.com/users/5665723/kwusf"
            },
            "is_answered": true,
            "view_count": 46,
            "answer_count": 4,
            "score": 1,
            "last_activity_date": 1524185662,
            "creation_date": 1524157080,
            "last_edit_date": 1524157709,
            "question_id": 49926447,
            "body_markdown": "I have data in 2 columns. The data in column B is comma delimited. I need each instance to appear on a new row while retaining it&#39;s original ID from column A. I also need the data in 3 columns so Name is in B and Number in C. It appears as so:\r\n\r\n\r\n----------\r\n\r\n\r\n&gt;&gt;A--------B\r\n\r\n&gt;&gt;1--------Sam Jones, 1 hours, Chris Bacon, 2 hours  \r\n2--------John Jacob, 3 hours  \r\n3--------John Hancock, 4 hours, Brian Smith, .5 hours\r\n\r\n\r\n----------\r\nI am able to get it as such using my code below:\r\n\r\n&gt;&gt;A--------B\r\n\r\n&gt;&gt;1--------Sam Jones, 1  \r\n1--------Chris Bacon, 2 hours  \r\n2--------John Jacob, 3 hours  \r\n3--------John Hancock, 4  \r\n3--------Brian Smith, .5 hours\r\n\r\n\r\n----------\r\n\r\n\r\nI need it to be: (notice last value in string also has hours removed when added to new line)\r\n\r\n&gt;&gt;A---------B------------------------C  \r\n1---------Sam Jones-----------1  \r\n1---------Chris Bacon----------2  \r\n2---------John Jacob-----------3  \r\n3---------John Hancock-------4  \r\n3---------Brian Smith----------.5  \r\n\r\nI have the following code started: (I can&#39;t manage to remove the &quot;hours&quot; from the last person in each delimited string and I can&#39;t get it into 3 columns)\r\n        \r\n    Sub splitByColB()  \r\n      Dim r As Range, i As Long, ar  \r\n      Set r = Worksheets(&quot;Sheet1&quot;).Range(&quot;B2&quot;).End(xlDown)  \r\n      Do While r.Row &gt; 1  \r\n        ar = Split(r.Value, &quot; hours, &quot;)  \r\n        If UBound(ar) &gt;= 0 Then r.Value = ar(0)  \r\n        For i = UBound(ar) To 1 Step -1  \r\n          r.EntireRow.Copy  \r\n          r.Offset(1).EntireRow.Insert  \r\n          r.Offset(1).Value = ar(i)  \r\n        Next  \r\n        Set r = r.Offset(-1)  \r\n      Loop  \r\n    End Sub  \r\n",
            "link": "https://stackoverflow.com/questions/49926447/splitting-and-replacing-delimited-strings-into-new-rows-using-vba-in-excel",
            "title": "Splitting and replacing delimited strings into new rows using VBA in Excel",
            "body": "<p>I have data in 2 columns. The data in column B is comma delimited. I need each instance to appear on a new row while retaining it's original ID from column A. I also need the data in 3 columns so Name is in B and Number in C. It appears as so:</p>\n\n<hr>\n\n<blockquote>\n  <blockquote>\n    <p>A--------B</p>\n    \n    <p>1--------Sam Jones, 1 hours, Chris Bacon, 2 hours<br>\n    2--------John Jacob, 3 hours<br>\n    3--------John Hancock, 4 hours, Brian Smith, .5 hours</p>\n  </blockquote>\n</blockquote>\n\n<hr>\n\n<p>I am able to get it as such using my code below:</p>\n\n<blockquote>\n  <blockquote>\n    <p>A--------B</p>\n    \n    <p>1--------Sam Jones, 1<br>\n    1--------Chris Bacon, 2 hours<br>\n    2--------John Jacob, 3 hours<br>\n    3--------John Hancock, 4<br>\n    3--------Brian Smith, .5 hours</p>\n  </blockquote>\n</blockquote>\n\n<hr>\n\n<p>I need it to be: (notice last value in string also has hours removed when added to new line)</p>\n\n<blockquote>\n  <blockquote>\n    <p>A---------B------------------------C<br>\n    1---------Sam Jones-----------1<br>\n    1---------Chris Bacon----------2<br>\n    2---------John Jacob-----------3<br>\n    3---------John Hancock-------4<br>\n    3---------Brian Smith----------.5  </p>\n  </blockquote>\n</blockquote>\n\n<p>I have the following code started: (I can't manage to remove the \"hours\" from the last person in each delimited string and I can't get it into 3 columns)</p>\n\n<pre><code>Sub splitByColB()  \n  Dim r As Range, i As Long, ar  \n  Set r = Worksheets(\"Sheet1\").Range(\"B2\").End(xlDown)  \n  Do While r.Row &gt; 1  \n    ar = Split(r.Value, \" hours, \")  \n    If UBound(ar) &gt;= 0 Then r.Value = ar(0)  \n    For i = UBound(ar) To 1 Step -1  \n      r.EntireRow.Copy  \n      r.Offset(1).EntireRow.Insert  \n      r.Offset(1).Value = ar(i)  \n    Next  \n    Set r = r.Offset(-1)  \n  Loop  \nEnd Sub  \n</code></pre>\n"
        },
        {
            "tags": [
                "c#",
                ".net",
                "entity-framework",
                "core"
            ],
            "owner": {
                "reputation": 52,
                "user_id": 9018838,
                "user_type": "registered",
                "accept_rate": 0,
                "profile_image": "https://www.gravatar.com/avatar/4c5a29cd6fe7038e415c3d7ac1a828d9?s=128&d=identicon&r=PG&f=1",
                "display_name": "JaeWang Lee",
                "link": "https://stackoverflow.com/users/9018838/jaewang-lee"
            },
            "is_answered": false,
            "view_count": 22,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185660,
            "creation_date": 1524146057,
            "last_edit_date": 1524185660,
            "question_id": 49922907,
            "body_markdown": "    public class LetterModel\r\n    {\r\n        public int LetterId { get; set; }\r\n        public ICollection&lt;AttachmentModel&gt; Attachments { get; set; }\r\n    }\r\n\r\n    public class AttachmentModel\r\n    {\r\n        public LetterModel Letter { get; set; }\r\n    }\r\n\r\n    modelBuilder.Entity&lt;LetterModel&gt;()\r\n       .HasMany(l =&gt; l.Attachments)\r\n       .WithOne(a =&gt; a.Letter)\r\n       .HasForeignKey(a =&gt; a.LetterId);\r\n\r\n    public async Task&lt;IEnumerable&lt;LetterModel&gt;&gt; Get()\r\n        {\r\n            var data = await _db.Letters\r\n                .Select(l =&gt; new LetterModel\r\n                {\r\n                    Id = l.Id,\r\n                    Attachments = l.Attachments,\r\n                }).ToListAsync();\r\n\r\n            return data;\r\n        }\r\n\r\n        [HttpGet(&quot;&quot;)]\r\n        public async Task&lt;IActionResult&gt; Get()\r\n        {\r\n            var letters = await _query.Get();\r\n            if (letters == null)\r\n            {\r\n                return NotFound();\r\n            }\r\n\r\n            return Ok(letters);\r\n        }\r\n\r\n        CONSTRAINT [FK_Attachments_Letters_LetterId] FOREIGN KEY ([LetterId])                                      \r\n         REFERENCES [dbo].[Letters] ([Id]) ON DELETE CASCADE\r\n        );\r\n\r\n\r\n        GO\r\n        CREATE NONCLUSTERED INDEX [IX_Attachments_LetterId]\r\n            ON [dbo].[Attachments]([LetterId] ASC);\r\n\r\nI coded above like that\r\nI have already used one-one relationship using HasOne and WithOne and HasForeignKey\r\n\r\nBut, When I use HasMany like that using HttpGet,\r\nMy Swagger shows **AbortError: The operation was aborted.**\r\n\r\nHow I solve this? Please help me",
            "link": "https://stackoverflow.com/questions/49922907/aborterror-when-using-hasmany-in-net-core",
            "title": "AbortError When Using HasMany in .NET Core",
            "body": "<pre><code>public class LetterModel\n{\n    public int LetterId { get; set; }\n    public ICollection&lt;AttachmentModel&gt; Attachments { get; set; }\n}\n\npublic class AttachmentModel\n{\n    public LetterModel Letter { get; set; }\n}\n\nmodelBuilder.Entity&lt;LetterModel&gt;()\n   .HasMany(l =&gt; l.Attachments)\n   .WithOne(a =&gt; a.Letter)\n   .HasForeignKey(a =&gt; a.LetterId);\n\npublic async Task&lt;IEnumerable&lt;LetterModel&gt;&gt; Get()\n    {\n        var data = await _db.Letters\n            .Select(l =&gt; new LetterModel\n            {\n                Id = l.Id,\n                Attachments = l.Attachments,\n            }).ToListAsync();\n\n        return data;\n    }\n\n    [HttpGet(\"\")]\n    public async Task&lt;IActionResult&gt; Get()\n    {\n        var letters = await _query.Get();\n        if (letters == null)\n        {\n            return NotFound();\n        }\n\n        return Ok(letters);\n    }\n\n    CONSTRAINT [FK_Attachments_Letters_LetterId] FOREIGN KEY ([LetterId])                                      \n     REFERENCES [dbo].[Letters] ([Id]) ON DELETE CASCADE\n    );\n\n\n    GO\n    CREATE NONCLUSTERED INDEX [IX_Attachments_LetterId]\n        ON [dbo].[Attachments]([LetterId] ASC);\n</code></pre>\n\n<p>I coded above like that\nI have already used one-one relationship using HasOne and WithOne and HasForeignKey</p>\n\n<p>But, When I use HasMany like that using HttpGet,\nMy Swagger shows <strong>AbortError: The operation was aborted.</strong></p>\n\n<p>How I solve this? Please help me</p>\n"
        },
        {
            "tags": [
                "java",
                "eclipse",
                "swing",
                "debugging"
            ],
            "owner": {
                "reputation": 83,
                "user_id": 3858961,
                "user_type": "registered",
                "accept_rate": 20,
                "profile_image": "https://www.gravatar.com/avatar/a53e86ea533e33555a9b508480e8ef82?s=128&d=identicon&r=PG&f=1",
                "display_name": "Maverick",
                "link": "https://stackoverflow.com/users/3858961/maverick"
            },
            "is_answered": true,
            "view_count": 2477,
            "closed_date": 1524197153,
            "answer_count": 3,
            "score": 1,
            "last_activity_date": 1524185634,
            "creation_date": 1405903745,
            "last_edit_date": 1405909496,
            "question_id": 24856333,
            "body_markdown": "i am new to java and eclipse. I am facing problem in debug mode. When I run my program in debug, it run successfully. But step into, step over buttons appear disable. Due to this issue I am not able to debug my program.\r\n\r\nthanks ",
            "link": "https://stackoverflow.com/questions/24856333/step-into-and-step-over-button-in-java-eclipse-disable",
            "closed_reason": "unclear what you&#39;re asking",
            "title": "step into and step over button in java eclipse disable",
            "body": "<p>i am new to java and eclipse. I am facing problem in debug mode. When I run my program in debug, it run successfully. But step into, step over buttons appear disable. Due to this issue I am not able to debug my program.</p>\n\n<p>thanks </p>\n"
        },
        {
            "tags": [
                "devops",
                "google-cloud-iot"
            ],
            "owner": {
                "reputation": 68,
                "user_id": 6597774,
                "user_type": "registered",
                "accept_rate": 60,
                "profile_image": "https://www.gravatar.com/avatar/a93c8ed3021c54919ce95926fa1fe2e6?s=128&d=identicon&r=PG&f=1",
                "display_name": "jlyh",
                "link": "https://stackoverflow.com/users/6597774/jlyh"
            },
            "is_answered": false,
            "view_count": 20,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185619,
            "creation_date": 1523789891,
            "question_id": 49841152,
            "body_markdown": "I have some issues trying to access google iot-core API from my remote server (hosted on Digital Ocean). The same piece of code could work on my local machine, and I ensured that even the service key was the same, yet there was no explicit message to point me in some general direction. \r\n\r\nI&#39;m not sure if this is an issue with the discovery API as I could get the proper response on my local machine using the curl command below but when i tried to do this on my remote server it returned a 404 error. Anyone experienced this before?\r\n\r\n    curl https://cloudiot.googleapis.com/$discovery/rest?version=v1",
            "link": "https://stackoverflow.com/questions/49841152/google-iot-core-v1-api-discovery-not-found",
            "title": "Google IoT-core v1 API discovery not found",
            "body": "<p>I have some issues trying to access google iot-core API from my remote server (hosted on Digital Ocean). The same piece of code could work on my local machine, and I ensured that even the service key was the same, yet there was no explicit message to point me in some general direction. </p>\n\n<p>I'm not sure if this is an issue with the discovery API as I could get the proper response on my local machine using the curl command below but when i tried to do this on my remote server it returned a 404 error. Anyone experienced this before?</p>\n\n<pre><code>curl https://cloudiot.googleapis.com/$discovery/rest?version=v1\n</code></pre>\n"
        },
        {
            "tags": [
                "dask",
                "dask-distributed"
            ],
            "owner": {
                "reputation": 16,
                "user_id": 9468602,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-XdUIqdMkCWA/AAAAAAAAAAI/AAAAAAAAAAA/4252rscbv5M/photo.jpg?sz=128",
                "display_name": "Ameet Shah",
                "link": "https://stackoverflow.com/users/9468602/ameet-shah"
            },
            "is_answered": false,
            "view_count": 15,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185607,
            "creation_date": 1524185607,
            "question_id": 49932282,
            "body_markdown": "We are running dask scheduler on windows VM and memory utilization gradually increases till system memory usage reaches 98%. We then have to restart the scheduler as else we receive timeouts from workers trying to connect. This does take a few days and our allocated memory for VM is 16GB.\r\n\r\nIs there any setting we need to do where the utilization of the memory by scheduler gets capped to certain amount. We are currently on distributed &#39;1.21.3&#39; and dask &#39;0.17.1&#39;\r\n\r\nUnlike the example in link - https://stackoverflow.com/questions/46567997/slowly-increasing-memory-usage-of-dask-sheduler the memory increase does not stabilize but slowly keeps increasing with job executions. ",
            "link": "https://stackoverflow.com/questions/49932282/dask-scheduler-gradual-memory-increase-causing-it-to-fail",
            "title": "Dask scheduler gradual memory increase causing it to fail",
            "body": "<p>We are running dask scheduler on windows VM and memory utilization gradually increases till system memory usage reaches 98%. We then have to restart the scheduler as else we receive timeouts from workers trying to connect. This does take a few days and our allocated memory for VM is 16GB.</p>\n\n<p>Is there any setting we need to do where the utilization of the memory by scheduler gets capped to certain amount. We are currently on distributed '1.21.3' and dask '0.17.1'</p>\n\n<p>Unlike the example in link - <a href=\"https://stackoverflow.com/questions/46567997/slowly-increasing-memory-usage-of-dask-sheduler\">Slowly increasing memory usage of Dask Sheduler</a> the memory increase does not stabilize but slowly keeps increasing with job executions. </p>\n"
        },
        {
            "tags": [
                ".net",
                "signalr",
                "webrtc"
            ],
            "owner": {
                "reputation": 591,
                "user_id": 1798757,
                "user_type": "registered",
                "accept_rate": 85,
                "profile_image": "https://www.gravatar.com/avatar/d487e672b401a0bd5d7a78ced8a20c93?s=128&d=identicon&r=PG",
                "display_name": "ikwillem",
                "link": "https://stackoverflow.com/users/1798757/ikwillem"
            },
            "is_answered": false,
            "view_count": 9,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185590,
            "creation_date": 1524185590,
            "question_id": 49932280,
            "body_markdown": "I&#39;m currently developing a conferencing video web application. I decided to go the WebRTC route, but it is still a bit unclear to me how to use it properly.\r\n\r\nMy application is made in ASP.NET MVC and I also added SignalR for chat/notification functionality. I need to make a live connection between the users and also record the stream of the speaker. The recording of the speaker now works.\r\n\r\nI made a page that starts a mediarecorder to capture the users webcam, then I attached to the ondataavailable event to capture the webcam data:\r\n\r\n    mediaRecorder.ondataavailable = function (e) {\r\n\r\n        console.log(&#39;Data received: &#39;, e);\r\n\r\n        // Increase index\r\n        streamindex++;\r\n\r\n        // Save client dates\r\n        var stopdate = new Date();\r\n        var startdate = olddate;\r\n        olddate = stopdate;\r\n\r\n        // Format dates\r\n        var startdatestr =\r\n            startdate.getFullYear() + &quot;-&quot; +\r\n            startdate.getMonth() + &quot;-&quot; +\r\n            startdate.getDate() + &quot; &quot; +\r\n            startdate.getHours() + &#39;:&#39; +\r\n            startdate.getMinutes() + &#39;:&#39; +\r\n            startdate.getSeconds() + &#39;.&#39; +\r\n            startdate.getMilliseconds();\r\n\r\n        var stopdatestr =\r\n            stopdate.getFullYear() + &quot;-&quot; +\r\n            stopdate.getMonth() + &quot;-&quot; +\r\n            stopdate.getDate() + &quot; &quot; +\r\n            stopdate.getHours() + &#39;:&#39; +\r\n            stopdate.getMinutes() + &#39;:&#39; +\r\n            stopdate.getSeconds() + &#39;.&#39; +\r\n            stopdate.getMilliseconds();\r\n\r\n        // Send data to server\r\n        $.ajax({\r\n            url: &#39;/Upload?StreamId=&#39; + streamid + &#39;&amp;Index=&#39; + streamindex + &#39;&amp;StartDate=&#39; + startdatestr + &#39;&amp;StopDate=&#39; + stopdatestr,\r\n            type: &#39;POST&#39;,\r\n            contentType: &#39;application/octet-stream&#39;,\r\n            data: e.data,\r\n            processData: false\r\n        });\r\n    };\r\n\r\nOn the server I save the data and this seems to work perfectly.\r\n\r\n\r\nNow I want to make a peer to peer connection between the clients, I&#39;ve looked online how to do this, but I found out I have to use STUN and TURN servers to create the connection. Since I use .NET I don&#39;t really like the idea of having to deploy nodejs servers. I found this code online:\r\n\r\n    function call() {\r\n        callButton.disabled = true;\r\n        hangupButton.disabled = false;\r\n        trace(&quot;Starting call&quot;);\r\n\r\n        if (localStream.getVideoTracks().length &gt; 0) {\r\n            trace(&#39;Using video device: &#39; + localStream.getVideoTracks()[0].label);\r\n        }\r\n        if (localStream.getAudioTracks().length &gt; 0) {\r\n            trace(&#39;Using audio device: &#39; + localStream.getAudioTracks()[0].label);\r\n        }\r\n\r\n        var servers = null;\r\n\r\n        localPeerConnection = new RTCPeerConnection(servers);\r\n        trace(&quot;Created local peer connection object localPeerConnection&quot;);\r\n        localPeerConnection.onicecandidate = gotLocalIceCandidate;\r\n\r\n        remotePeerConnection = new RTCPeerConnection(servers);\r\n        trace(&quot;Created remote peer connection object remotePeerConnection&quot;);\r\n        remotePeerConnection.onicecandidate = gotRemoteIceCandidate;\r\n        remotePeerConnection.onaddstream = gotRemoteStream;\r\n\r\n        localPeerConnection.addStream(localStream);\r\n        trace(&quot;Added localStream to localPeerConnection&quot;);\r\n        localPeerConnection.createOffer(gotLocalDescription, handleError);\r\n    }\r\n\r\n    function gotLocalDescription(description) {\r\n        localPeerConnection.setLocalDescription(description);\r\n        trace(&quot;Offer from localPeerConnection: \\n&quot; + description.sdp);\r\n        remotePeerConnection.setRemoteDescription(description);\r\n        remotePeerConnection.createAnswer(gotRemoteDescription, handleError);\r\n    }\r\n\r\n    function gotRemoteDescription(description) {\r\n        remotePeerConnection.setLocalDescription(description);\r\n        trace(&quot;Answer from remotePeerConnection: \\n&quot; + description.sdp);\r\n        localPeerConnection.setRemoteDescription(description);\r\n    }\r\n\r\n    function hangup() {\r\n        trace(&quot;Ending call&quot;);\r\n        localPeerConnection.close();\r\n        remotePeerConnection.close();\r\n        localPeerConnection = null;\r\n        remotePeerConnection = null;\r\n        hangupButton.disabled = true;\r\n        callButton.disabled = false;\r\n    }\r\n\r\n    function gotRemoteStream(event) {\r\n        remoteVideo.src = URL.createObjectURL(event.stream);\r\n        trace(&quot;Received remote stream&quot;);\r\n    }\r\n\r\n    function gotLocalIceCandidate(event) {\r\n        if (event.candidate) {\r\n            remotePeerConnection.addIceCandidate(new RTCIceCandidate(event.candidate));\r\n            trace(&quot;Local ICE candidate: \\n&quot; + event.candidate.candidate);\r\n        }\r\n    }\r\n\r\n    function gotRemoteIceCandidate(event) {\r\n        if (event.candidate) {\r\n            localPeerConnection.addIceCandidate(new RTCIceCandidate(event.candidate));\r\n            trace(&quot;Remote ICE candidate: \\n &quot; + event.candidate.candidate);\r\n        }\r\n    }\r\n\r\nThis example uses a local RTCPeerConnection to send the video/audio data. If I want to use this example online I should add the STUN / TURN servers to the &quot;servers&quot; variable. But since I have no STUN / TURN servers, is there any way I can make the WebRTC connection using SignalR?\r\n\r\n(SignalR allows me to send messages / data between the clients very easily)",
            "link": "https://stackoverflow.com/questions/49932280/webrtc-ice-stun-and-turn-but-cant-i-just-use-my-signalr-net-connection",
            "title": "WebRTC: ICE, STUN and TURN, but can&#39;t I just use my SignalR.NET connection?",
            "body": "<p>I'm currently developing a conferencing video web application. I decided to go the WebRTC route, but it is still a bit unclear to me how to use it properly.</p>\n\n<p>My application is made in ASP.NET MVC and I also added SignalR for chat/notification functionality. I need to make a live connection between the users and also record the stream of the speaker. The recording of the speaker now works.</p>\n\n<p>I made a page that starts a mediarecorder to capture the users webcam, then I attached to the ondataavailable event to capture the webcam data:</p>\n\n<pre><code>mediaRecorder.ondataavailable = function (e) {\n\n    console.log('Data received: ', e);\n\n    // Increase index\n    streamindex++;\n\n    // Save client dates\n    var stopdate = new Date();\n    var startdate = olddate;\n    olddate = stopdate;\n\n    // Format dates\n    var startdatestr =\n        startdate.getFullYear() + \"-\" +\n        startdate.getMonth() + \"-\" +\n        startdate.getDate() + \" \" +\n        startdate.getHours() + ':' +\n        startdate.getMinutes() + ':' +\n        startdate.getSeconds() + '.' +\n        startdate.getMilliseconds();\n\n    var stopdatestr =\n        stopdate.getFullYear() + \"-\" +\n        stopdate.getMonth() + \"-\" +\n        stopdate.getDate() + \" \" +\n        stopdate.getHours() + ':' +\n        stopdate.getMinutes() + ':' +\n        stopdate.getSeconds() + '.' +\n        stopdate.getMilliseconds();\n\n    // Send data to server\n    $.ajax({\n        url: '/Upload?StreamId=' + streamid + '&amp;Index=' + streamindex + '&amp;StartDate=' + startdatestr + '&amp;StopDate=' + stopdatestr,\n        type: 'POST',\n        contentType: 'application/octet-stream',\n        data: e.data,\n        processData: false\n    });\n};\n</code></pre>\n\n<p>On the server I save the data and this seems to work perfectly.</p>\n\n<p>Now I want to make a peer to peer connection between the clients, I've looked online how to do this, but I found out I have to use STUN and TURN servers to create the connection. Since I use .NET I don't really like the idea of having to deploy nodejs servers. I found this code online:</p>\n\n<pre><code>function call() {\n    callButton.disabled = true;\n    hangupButton.disabled = false;\n    trace(\"Starting call\");\n\n    if (localStream.getVideoTracks().length &gt; 0) {\n        trace('Using video device: ' + localStream.getVideoTracks()[0].label);\n    }\n    if (localStream.getAudioTracks().length &gt; 0) {\n        trace('Using audio device: ' + localStream.getAudioTracks()[0].label);\n    }\n\n    var servers = null;\n\n    localPeerConnection = new RTCPeerConnection(servers);\n    trace(\"Created local peer connection object localPeerConnection\");\n    localPeerConnection.onicecandidate = gotLocalIceCandidate;\n\n    remotePeerConnection = new RTCPeerConnection(servers);\n    trace(\"Created remote peer connection object remotePeerConnection\");\n    remotePeerConnection.onicecandidate = gotRemoteIceCandidate;\n    remotePeerConnection.onaddstream = gotRemoteStream;\n\n    localPeerConnection.addStream(localStream);\n    trace(\"Added localStream to localPeerConnection\");\n    localPeerConnection.createOffer(gotLocalDescription, handleError);\n}\n\nfunction gotLocalDescription(description) {\n    localPeerConnection.setLocalDescription(description);\n    trace(\"Offer from localPeerConnection: \\n\" + description.sdp);\n    remotePeerConnection.setRemoteDescription(description);\n    remotePeerConnection.createAnswer(gotRemoteDescription, handleError);\n}\n\nfunction gotRemoteDescription(description) {\n    remotePeerConnection.setLocalDescription(description);\n    trace(\"Answer from remotePeerConnection: \\n\" + description.sdp);\n    localPeerConnection.setRemoteDescription(description);\n}\n\nfunction hangup() {\n    trace(\"Ending call\");\n    localPeerConnection.close();\n    remotePeerConnection.close();\n    localPeerConnection = null;\n    remotePeerConnection = null;\n    hangupButton.disabled = true;\n    callButton.disabled = false;\n}\n\nfunction gotRemoteStream(event) {\n    remoteVideo.src = URL.createObjectURL(event.stream);\n    trace(\"Received remote stream\");\n}\n\nfunction gotLocalIceCandidate(event) {\n    if (event.candidate) {\n        remotePeerConnection.addIceCandidate(new RTCIceCandidate(event.candidate));\n        trace(\"Local ICE candidate: \\n\" + event.candidate.candidate);\n    }\n}\n\nfunction gotRemoteIceCandidate(event) {\n    if (event.candidate) {\n        localPeerConnection.addIceCandidate(new RTCIceCandidate(event.candidate));\n        trace(\"Remote ICE candidate: \\n \" + event.candidate.candidate);\n    }\n}\n</code></pre>\n\n<p>This example uses a local RTCPeerConnection to send the video/audio data. If I want to use this example online I should add the STUN / TURN servers to the \"servers\" variable. But since I have no STUN / TURN servers, is there any way I can make the WebRTC connection using SignalR?</p>\n\n<p>(SignalR allows me to send messages / data between the clients very easily)</p>\n"
        },
        {
            "tags": [
                "regex",
                "perl",
                "xregexp"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 9672304,
                "user_type": "registered",
                "profile_image": "https://lh3.googleusercontent.com/-XdUIqdMkCWA/AAAAAAAAAAI/AAAAAAAAAAA/4252rscbv5M/photo.jpg?sz=128",
                "display_name": "JO P",
                "link": "https://stackoverflow.com/users/9672304/jo-p"
            },
            "is_answered": false,
            "view_count": 64,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185555,
            "creation_date": 1524182676,
            "last_edit_date": 1524183200,
            "question_id": 49931959,
            "body_markdown": "I want to write a Perl regexp which gives a match if a string \r\n\r\n 1. begins with &quot;Reply-To: &quot;\r\n 2. continues with any number of characters\r\n 3. continues with only &quot;@domain1.fr&quot; or &quot;@domain2.fr\r\n 4. ends with any number of characters\r\n\r\nExamples :\r\n\r\n - Reply-To: abc def@domain1      must match\r\n - Reply-To: abc def@domain2      must match\r\n - Reply-To: abc def@domain3      must not match\r\n\r\n\r\n\r\nI tried\r\n\r\n    /^Reply-To: *(?!@domain1.fr|@domain2.fr)/ \r\n\r\nbut\r\n\r\n - Reply-To: abc def@domain1 doesn&#39;t match\r\n - Reply-To: abc def@domain2 doesn&#39;t matches\r\n\r\n\r\nThe problem is to exclude from the match domains that are **different from domain1 AND domain2**. Because, if I want to exclude from the match domains that are different only from domain1, I type\r\n\r\n    /^Reply-To: *(?!domain1.fr)/ \r\n\r\nand it works perfectly.\r\n",
            "link": "https://stackoverflow.com/questions/49931959/perl-regexp-for-reply-to-then-characters-then-only-domain1-fr-or-domain",
            "title": "Perl regexp for &quot;Reply-To: &quot; then characters then only &quot;@domain1.fr&quot; or &quot;@domain2.fr&quot;",
            "body": "<p>I want to write a Perl regexp which gives a match if a string </p>\n\n<ol>\n<li>begins with \"Reply-To: \"</li>\n<li>continues with any number of characters</li>\n<li>continues with only \"@domain1.fr\" or \"@domain2.fr</li>\n<li>ends with any number of characters</li>\n</ol>\n\n<p>Examples :</p>\n\n<ul>\n<li>Reply-To: abc def@domain1      must match</li>\n<li>Reply-To: abc def@domain2      must match</li>\n<li>Reply-To: abc def@domain3      must not match</li>\n</ul>\n\n<p>I tried</p>\n\n<pre><code>/^Reply-To: *(?!@domain1.fr|@domain2.fr)/ \n</code></pre>\n\n<p>but</p>\n\n<ul>\n<li>Reply-To: abc def@domain1 doesn't match</li>\n<li>Reply-To: abc def@domain2 doesn't matches</li>\n</ul>\n\n<p>The problem is to exclude from the match domains that are <strong>different from domain1 AND domain2</strong>. Because, if I want to exclude from the match domains that are different only from domain1, I type</p>\n\n<pre><code>/^Reply-To: *(?!domain1.fr)/ \n</code></pre>\n\n<p>and it works perfectly.</p>\n"
        },
        {
            "tags": [
                "nginx",
                "kubernetes"
            ],
            "owner": {
                "reputation": 78,
                "user_id": 6800389,
                "user_type": "registered",
                "accept_rate": 96,
                "profile_image": "https://www.gravatar.com/avatar/f9f28dc641e5ac8fe83f1094d298ae12?s=128&d=identicon&r=PG&f=1",
                "display_name": "Brando__",
                "link": "https://stackoverflow.com/users/6800389/brando"
            },
            "is_answered": false,
            "view_count": 20,
            "answer_count": 1,
            "score": 0,
            "last_activity_date": 1524185545,
            "creation_date": 1524171433,
            "question_id": 49930128,
            "body_markdown": "I am using the ingress-nginx system https://github.com/kubernetes/ingress-nginx. I&#39;m making extensive use of this project. Jenkins, Consul, Prometheus and more are working just fine using the exact same ingress config as appended to the end.\r\n\r\n\r\nI am able to access my vault pod directly through port-forwarding with kubectl. But when I attempt to access it over my nginx-ingress, I am returned a 503\r\n\r\n    kubectl port-forward vault-vault-f9778f86d-srr9n 8200:8200 -n vault\r\n\r\n      curl 127.0.0.1:8200/v1/1\r\n        {&quot;errors&quot;:[&quot;Vault is sealed&quot;]}\r\n\r\n    ➜  vault curl -L vault.me.com/v1/1\r\n    &lt;html&gt;\r\n    &lt;head&gt;&lt;title&gt;503 Service Temporarily Unavailable&lt;/title&gt;&lt;/head&gt;\r\n    &lt;body bgcolor=&quot;white&quot;&gt;\r\n    &lt;center&gt;&lt;h1&gt;503 Service Temporarily Unavailable&lt;/h1&gt;&lt;/center&gt;\r\n    &lt;hr&gt;&lt;center&gt;nginx/1.13.8&lt;/center&gt;\r\n    &lt;/body&gt;\r\n    &lt;/html&gt;\r\n\r\nLooking at the logs I see the following in response to the vault.me.com curl\r\n\r\n    10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:42:56 +0000] &quot;GET / HTTP/1.1&quot; 308 187 &quot;-&quot; &quot;curl/7.43.0&quot; 77 0.000 [] - - - -\r\n    10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:42:56 +0000] &quot;GET / HTTP/1.1&quot; 503 213 &quot;-&quot; &quot;curl/7.43.0&quot; 77 0.000 [] - - - -\r\n\r\nWhere as if I try to access my consul backend, I see the following.\r\n\r\n    10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:34 +0000] &quot;GET / HTTP/1.1&quot; 308 187 &quot;-&quot; &quot;curl/7.43.0&quot; 78 0.000 [consul-consul-consul-8500] - - - -\r\n    10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:39 +0000] &quot;GET / HTTP/1.1&quot; 308 187 &quot;-&quot; &quot;curl/7.43.0&quot; 78 0.000 [consul-consul-consul-8500] - - - -\r\n    10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:39 +0000] &quot;GET / HTTP/1.1&quot; 301 39 &quot;-&quot; &quot;curl/7.43.0&quot; 78 0.002 [consul-consul-consul-8500] 10.233.114.4:8500 39 0.002 301\r\n    10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:39 +0000] &quot;GET /ui/ HTTP/1.1&quot; 200 30178 &quot;-&quot; &quot;curl/7.43.0&quot; 81 0.001 [consul-consul-consul-8500] 10.233.82.19:8500 30178 0.001 200\r\n\r\nI&#39;m not entire sure whats going on, nor am I clear on how to debug this. I&#39;ve spent the two hours reading through the source on this tool, but I&#39;m not seeing much of anything.\r\n\r\nI read somewhere that any response that isn&#39;t 2xx or 3xx  fails nginx&#39;s backend health check and will be removed from the pool of backends to be routed to. That would manifest as the 503 that I am seeing. Its true that vault&#39;s / does return a non 2xx or 3xx so that would make sense.\r\n\r\n\r\n\r\n    vault curl -v 127.0.0.1:8200\r\n    * Rebuilt URL to: 127.0.0.1:8200/\r\n    *   Trying 127.0.0.1...\r\n    * Connected to 127.0.0.1 (127.0.0.1) port 8200 (#0)\r\n    &gt; GET / HTTP/1.1\r\n    &gt; Host: 127.0.0.1:8200\r\n    &gt; User-Agent: curl/7.43.0\r\n    &gt; Accept: */*\r\n    &gt;\r\n    &lt; HTTP/1.1 404 Not Found\r\n    &lt; Cache-Control: no-store\r\n    &lt; Content-Type: text/plain; charset=utf-8\r\n    &lt; X-Content-Type-Options: nosniff\r\n    &lt; Date: Thu, 19 Apr 2018 20:46:40 GMT\r\n    &lt; Content-Length: 19\r\n    &lt;\r\n    404 page not found\r\n\r\n**Is there a way to change the URL that nginx tries to healthcheck, or disable the healthchecking entirely**\r\n\r\nhttps://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/\r\nFrom nginx&#39;s docs it says if there is a single server in a backend group, it should never be getting marked as unavailable, regardless of return code.\r\n\r\nWhats going on?\r\n\r\n&gt; Note that if there is only a single server in a group, the\r\n&gt; fail_timeout and max_fails parameters are ignored and the server is\r\n&gt; never marked unavailable.\r\n\r\n    k get svc -n vault\r\n    NAME          TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE\r\n    vault-vault   ClusterIP   10.233.47.151   &lt;none&gt;        8200/TCP   1h\r\n\r\n    apiVersion: extensions/v1beta1\r\n    kind: Ingress\r\n    metadata:\r\n      annotations:\r\n        kubernetes.io/ingress.class: nginx-internal\r\n      name: vault\r\n      namespace: vault\r\n    spec:\r\n      rules:\r\n      - host: vault.me.com\r\n        http:\r\n          paths:\r\n          - backend:\r\n              serviceName: vault-vault\r\n              servicePort: 8200\r\n            path: /\r\n      tls:\r\n      - hosts:\r\n        - me.com\r\n        - vault.me.com\r\n        secretName: wildcard-secret\r\n\r\n\r\n\r\n\r\n",
            "link": "https://stackoverflow.com/questions/49930128/ingress-nginx-for-kubernetes-for-vault-suspected-lack-of-successful-healthcheck",
            "title": "Ingress-nginx for kubernetes for vault. Suspected lack of successful healthchecking",
            "body": "<p>I am using the ingress-nginx system <a href=\"https://github.com/kubernetes/ingress-nginx\" rel=\"nofollow noreferrer\">https://github.com/kubernetes/ingress-nginx</a>. I'm making extensive use of this project. Jenkins, Consul, Prometheus and more are working just fine using the exact same ingress config as appended to the end.</p>\n\n<p>I am able to access my vault pod directly through port-forwarding with kubectl. But when I attempt to access it over my nginx-ingress, I am returned a 503</p>\n\n<pre><code>kubectl port-forward vault-vault-f9778f86d-srr9n 8200:8200 -n vault\n\n  curl 127.0.0.1:8200/v1/1\n    {\"errors\":[\"Vault is sealed\"]}\n\n➜  vault curl -L vault.me.com/v1/1\n&lt;html&gt;\n&lt;head&gt;&lt;title&gt;503 Service Temporarily Unavailable&lt;/title&gt;&lt;/head&gt;\n&lt;body bgcolor=\"white\"&gt;\n&lt;center&gt;&lt;h1&gt;503 Service Temporarily Unavailable&lt;/h1&gt;&lt;/center&gt;\n&lt;hr&gt;&lt;center&gt;nginx/1.13.8&lt;/center&gt;\n&lt;/body&gt;\n&lt;/html&gt;\n</code></pre>\n\n<p>Looking at the logs I see the following in response to the vault.me.com curl</p>\n\n<pre><code>10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:42:56 +0000] \"GET / HTTP/1.1\" 308 187 \"-\" \"curl/7.43.0\" 77 0.000 [] - - - -\n10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:42:56 +0000] \"GET / HTTP/1.1\" 503 213 \"-\" \"curl/7.43.0\" 77 0.000 [] - - - -\n</code></pre>\n\n<p>Where as if I try to access my consul backend, I see the following.</p>\n\n<pre><code>10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:34 +0000] \"GET / HTTP/1.1\" 308 187 \"-\" \"curl/7.43.0\" 78 0.000 [consul-consul-consul-8500] - - - -\n10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:39 +0000] \"GET / HTTP/1.1\" 308 187 \"-\" \"curl/7.43.0\" 78 0.000 [consul-consul-consul-8500] - - - -\n10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:39 +0000] \"GET / HTTP/1.1\" 301 39 \"-\" \"curl/7.43.0\" 78 0.002 [consul-consul-consul-8500] 10.233.114.4:8500 39 0.002 301\n10.233.104.128 - [10.233.104.128] - - [19/Apr/2018:20:43:39 +0000] \"GET /ui/ HTTP/1.1\" 200 30178 \"-\" \"curl/7.43.0\" 81 0.001 [consul-consul-consul-8500] 10.233.82.19:8500 30178 0.001 200\n</code></pre>\n\n<p>I'm not entire sure whats going on, nor am I clear on how to debug this. I've spent the two hours reading through the source on this tool, but I'm not seeing much of anything.</p>\n\n<p>I read somewhere that any response that isn't 2xx or 3xx  fails nginx's backend health check and will be removed from the pool of backends to be routed to. That would manifest as the 503 that I am seeing. Its true that vault's / does return a non 2xx or 3xx so that would make sense.</p>\n\n<pre><code>vault curl -v 127.0.0.1:8200\n* Rebuilt URL to: 127.0.0.1:8200/\n*   Trying 127.0.0.1...\n* Connected to 127.0.0.1 (127.0.0.1) port 8200 (#0)\n&gt; GET / HTTP/1.1\n&gt; Host: 127.0.0.1:8200\n&gt; User-Agent: curl/7.43.0\n&gt; Accept: */*\n&gt;\n&lt; HTTP/1.1 404 Not Found\n&lt; Cache-Control: no-store\n&lt; Content-Type: text/plain; charset=utf-8\n&lt; X-Content-Type-Options: nosniff\n&lt; Date: Thu, 19 Apr 2018 20:46:40 GMT\n&lt; Content-Length: 19\n&lt;\n404 page not found\n</code></pre>\n\n<p><strong>Is there a way to change the URL that nginx tries to healthcheck, or disable the healthchecking entirely</strong></p>\n\n<p><a href=\"https://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/\" rel=\"nofollow noreferrer\">https://docs.nginx.com/nginx/admin-guide/load-balancer/http-health-check/</a>\nFrom nginx's docs it says if there is a single server in a backend group, it should never be getting marked as unavailable, regardless of return code.</p>\n\n<p>Whats going on?</p>\n\n<blockquote>\n  <p>Note that if there is only a single server in a group, the\n  fail_timeout and max_fails parameters are ignored and the server is\n  never marked unavailable.</p>\n</blockquote>\n\n<pre><code>k get svc -n vault\nNAME          TYPE        CLUSTER-IP      EXTERNAL-IP   PORT(S)    AGE\nvault-vault   ClusterIP   10.233.47.151   &lt;none&gt;        8200/TCP   1h\n\napiVersion: extensions/v1beta1\nkind: Ingress\nmetadata:\n  annotations:\n    kubernetes.io/ingress.class: nginx-internal\n  name: vault\n  namespace: vault\nspec:\n  rules:\n  - host: vault.me.com\n    http:\n      paths:\n      - backend:\n          serviceName: vault-vault\n          servicePort: 8200\n        path: /\n  tls:\n  - hosts:\n    - me.com\n    - vault.me.com\n    secretName: wildcard-secret\n</code></pre>\n"
        },
        {
            "tags": [
                "unity3d",
                "networking"
            ],
            "owner": {
                "reputation": 1,
                "user_id": 9189784,
                "user_type": "registered",
                "profile_image": "https://lh5.googleusercontent.com/-c7fICLBOCG0/AAAAAAAAAAI/AAAAAAAAABY/PahT35OQUpI/photo.jpg?sz=128",
                "display_name": "William Ehrenberg",
                "link": "https://stackoverflow.com/users/9189784/william-ehrenberg"
            },
            "is_answered": false,
            "view_count": 20,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185516,
            "creation_date": 1524185516,
            "question_id": 49932275,
            "body_markdown": "I am making a multiplayer MMO with unity and I am stuck on syncing up enemy spawning between clients. Basically the enemies spawn as intended, but only on the user that spawned them. Here is my code (Sorry it is a bit long, just wanted to include any information that could be relevant, as the is my first time with networking):\r\n\r\n    using System.Collections;\r\n    using System.Collections.Generic;\r\n    using UnityEngine;\r\n    using UnityEngine.Networking;\r\n\r\n    [System.Serializable]\r\n    public class EnemySpawn {\r\n\t    public GameObject prefab;\r\n\t    public int difficulty;\r\n\t    public int rarity;\r\n\t    [HideInInspector()]\r\n\t    public GameObject instance;\r\n    }\r\n \r\n    public class EnemySpawner : NetworkBehaviour {\r\n    \r\n    \tpublic EnemySpawn[] enemys;\r\n\t    public int maxDiff = 8;\r\n\t    public int spawnDistMin = 15;\r\n    \tpublic int spawnDistMax = 35;\r\n\r\n\t    public int spawnLvlDist = 30;\r\n\r\n    \tprivate int diffOnScreen;\r\n    \tprivate EnemySpawn enemyToSpawn;\r\n    \tpublic List&lt;EnemySpawn&gt; enemysOnScreen = new List&lt;EnemySpawn&gt;();\r\n\r\n\t    void Start() {\r\n\t    \tStartCoroutine (Spawn());\r\n\t    }\r\n\r\n\t    void Update() {\r\n\t    \tList&lt;EnemySpawn&gt; newEnemysOnScreen = new List&lt;EnemySpawn&gt;();\r\n\t    \tforeach(EnemySpawn c in enemysOnScreen) {\r\n\t\t    \tif (c.instance == null) {\r\n\t    \t\t\tdiffOnScreen -= c.difficulty;\r\n\t    \t\t} else {\r\n\t\t    \t\tnewEnemysOnScreen.Add (c);\r\n\t\t    \t}\r\n\t        }\r\n\t    \tenemysOnScreen = newEnemysOnScreen;\r\n\t    }\r\n    \r\n\r\n         //IMPORTANT PART!!! \\/\\/\\/\r\n\t    [Command]\r\n\t    void CmdSpawn(Vector3 pos) {\r\n\t\tGameObject spawning = Instantiate (enemyToSpawn.prefab, pos, Quaternion.identity);\r\n\t\tNetworkServer.Spawn (spawning);\r\n\t\tdiffOnScreen += enemyToSpawn.difficulty;\r\n\t\tenemysOnScreen.Insert (0,enemyToSpawn);\r\n\t\tenemysOnScreen [0].instance = spawning;\r\n\r\n\t\tspawning.GetComponent&lt;MonsterMain&gt;().lvl = Mathf.RoundToInt(spawning.transform.position.magnitude / spawnLvlDist);\r\n\t}\r\n\r\n\tIEnumerator Spawn() {\r\n\t\twhile (true) {\r\n\t\t\tint n = 0;\r\n\t\t\twhile (n != 1) {\r\n\t\t\t\tenemyToSpawn = enemys [Random.Range (0, enemys.Length)];\r\n\t\t\t\tn = Random.Range (1, enemyToSpawn.rarity);\r\n\t\t\t}\r\n\t\t\twhile (maxDiff - diffOnScreen &lt;= enemyToSpawn.difficulty) {\r\n\t\t\t\tyield return null;\r\n\t\t\t}\r\n\t\t\tyield return null;\r\n\t\t\tTransform player = Player.localPlayer.transform;\r\n\t\t\tVector2 randomOffset = Random.insideUnitCircle.normalized * Random.Range (spawnDistMin, spawnDistMax);\r\n\r\n\r\n\t\t\tCmdSpawn(player.position + new Vector3(randomOffset.x,randomOffset.y,0));\r\n\r\n\t\t}\r\n\t}\r\n\r\n}\r\n\r\nI am using Unity 2017.1.3.3p3, and I am building this to IOS. It is a 2d game and I am using unity&#39;s build in networking system. Again everything works except that the enemies only spawn on one screen. I am connecting one phone to the editor. Any help is appreciated. thank you!",
            "link": "https://stackoverflow.com/questions/49932275/unity-networking-enemy-spawning-is-not-working",
            "title": "Unity Networking enemy spawning is not working",
            "body": "<p>I am making a multiplayer MMO with unity and I am stuck on syncing up enemy spawning between clients. Basically the enemies spawn as intended, but only on the user that spawned them. Here is my code (Sorry it is a bit long, just wanted to include any information that could be relevant, as the is my first time with networking):</p>\n\n<pre><code>using System.Collections;\nusing System.Collections.Generic;\nusing UnityEngine;\nusing UnityEngine.Networking;\n\n[System.Serializable]\npublic class EnemySpawn {\n    public GameObject prefab;\n    public int difficulty;\n    public int rarity;\n    [HideInInspector()]\n    public GameObject instance;\n}\n\npublic class EnemySpawner : NetworkBehaviour {\n\n    public EnemySpawn[] enemys;\n    public int maxDiff = 8;\n    public int spawnDistMin = 15;\n    public int spawnDistMax = 35;\n\n    public int spawnLvlDist = 30;\n\n    private int diffOnScreen;\n    private EnemySpawn enemyToSpawn;\n    public List&lt;EnemySpawn&gt; enemysOnScreen = new List&lt;EnemySpawn&gt;();\n\n    void Start() {\n        StartCoroutine (Spawn());\n    }\n\n    void Update() {\n        List&lt;EnemySpawn&gt; newEnemysOnScreen = new List&lt;EnemySpawn&gt;();\n        foreach(EnemySpawn c in enemysOnScreen) {\n            if (c.instance == null) {\n                diffOnScreen -= c.difficulty;\n            } else {\n                newEnemysOnScreen.Add (c);\n            }\n        }\n        enemysOnScreen = newEnemysOnScreen;\n    }\n\n\n     //IMPORTANT PART!!! \\/\\/\\/\n    [Command]\n    void CmdSpawn(Vector3 pos) {\n    GameObject spawning = Instantiate (enemyToSpawn.prefab, pos, Quaternion.identity);\n    NetworkServer.Spawn (spawning);\n    diffOnScreen += enemyToSpawn.difficulty;\n    enemysOnScreen.Insert (0,enemyToSpawn);\n    enemysOnScreen [0].instance = spawning;\n\n    spawning.GetComponent&lt;MonsterMain&gt;().lvl = Mathf.RoundToInt(spawning.transform.position.magnitude / spawnLvlDist);\n}\n\nIEnumerator Spawn() {\n    while (true) {\n        int n = 0;\n        while (n != 1) {\n            enemyToSpawn = enemys [Random.Range (0, enemys.Length)];\n            n = Random.Range (1, enemyToSpawn.rarity);\n        }\n        while (maxDiff - diffOnScreen &lt;= enemyToSpawn.difficulty) {\n            yield return null;\n        }\n        yield return null;\n        Transform player = Player.localPlayer.transform;\n        Vector2 randomOffset = Random.insideUnitCircle.normalized * Random.Range (spawnDistMin, spawnDistMax);\n\n\n        CmdSpawn(player.position + new Vector3(randomOffset.x,randomOffset.y,0));\n\n    }\n}\n</code></pre>\n\n<p>}</p>\n\n<p>I am using Unity 2017.1.3.3p3, and I am building this to IOS. It is a 2d game and I am using unity's build in networking system. Again everything works except that the enemies only spawn on one screen. I am connecting one phone to the editor. Any help is appreciated. thank you!</p>\n"
        },
        {
            "tags": [
                "c++11",
                "infix-operator"
            ],
            "owner": {
                "reputation": 15,
                "user_id": 9319233,
                "user_type": "registered",
                "accept_rate": 0,
                "profile_image": "https://lh4.googleusercontent.com/-d6dHZYzM5uY/AAAAAAAAAAI/AAAAAAAACKw/B__GWVkTauk/photo.jpg?sz=128",
                "display_name": "Jack Bunzow",
                "link": "https://stackoverflow.com/users/9319233/jack-bunzow"
            },
            "is_answered": false,
            "view_count": 15,
            "answer_count": 0,
            "score": 0,
            "last_activity_date": 1524185503,
            "creation_date": 1524185503,
            "question_id": 49932274,
            "body_markdown": "I&#39;m working on a infix calculator for one of my classes and I&#39;m fairly new to stacks and such and I have hit a wall when its comes to implementing the execution of the operations of the values. The function used for executing only receives a operator and one value parameter, so I&#39;m lost as to how to perform the operations on a value when I need two. I am basing my code off of pseudo-code that was supplied by the professor ([This is a picture of it][1]). If anyone could point me in the right direction I&#39;d appreciate it. This is what I have.\r\n\r\n\r\n    int main()\r\n    {\r\n\t    stack&lt;char&gt; op;\r\n\t    stack&lt;int&gt;  val;\r\n\r\n\t    \tint result;\r\n\t    \tchar ch;\r\n\t    \t\r\n\t    \tcin &gt;&gt; ch;\r\n\t\r\n\t    \twhile (ch != &#39;Q&#39;) //Once it reads &#39;Q&#39; the program quits\r\n\t    \t{\r\n\t    \t\tswitch (ch)\r\n\t\t    \t{\r\n\t\t    \t\tcase &#39;0&#39;: case &#39;1&#39;: case &#39;2&#39;: case &#39;3&#39;: case &#39;4&#39;:\r\n\t    \t\t\tcase &#39;5&#39;: case &#39;6&#39;: case &#39;7&#39;: case &#39;8&#39;: case &#39;9&#39;:\r\n\t\t    \t\t\tval.push(ch);\r\n\t\t    \t\t\tbreak;\r\n\t\t\t\r\n\t\t\t    \tcase &#39;(&#39;:\r\n\t\t\t\t    \top.push(ch);\r\n\t\t\t\t    \tbreak;\r\n\t\t\t\t\r\n\t\t\t    \tcase &#39;+&#39;: case &#39;-&#39;: case &#39;*&#39;: case &#39;/&#39;:\r\n\t\t\t\t    \tif (op.empty())\r\n\t\t\t\t    \t{\r\n\t\t\t\t    \t\top.push(ch);\r\n\t\t\t\t    \t}\r\n\t\t\t\t\t    else if (Precedence(ch) &gt; Precedence(op.top()))\r\n\t\t\t\t\t    {\r\n\t\t\t\t\t    \top.push(ch);\r\n\t\t\t\t    \t}\r\n\t\t\t\t    \telse\r\n\t\t\t\t    \t{\r\n\t\t\t\t    \t\twhile ( (!op.empty()) &amp;&amp; (Precedence(ch) &lt;= \t    Precedence(op.top())) )\r\n\t\t\t\t\t    \t{\r\n\t\t\t\t\t    \t\tExecute(op, val);\r\n\t\t\t\t\t    \t}\r\n\t\t\t\t\t    \top.push(ch);\r\n\t\t\t\t    \t}\r\n\t\t\t\t    \tbreak;\r\n\t\t\t\t    \t\r\n\t\t\t    \tcase &#39;)&#39;:\r\n\t\t\t    \t{\r\n\t\t\t\t    \twhile (op.top() != &#39;(&#39;)\r\n\t\t\t\t\t    {\r\n\t\t\t\t    \t\tExecute(op, val);\r\n\t\t\t\t    \t}\r\n\t\t\t\t    \top.pop();\r\n\t\t\t\t    \tbreak;\r\n\t\t\t    \t}\r\n\t\t    \t}\r\n\t\t\r\n\t\t\t    cin &gt;&gt; ch;\r\n\t\t    }\r\n\t\r\n\t    \twhile (!op.empty())\r\n\t    \t{\r\n\t    \t\tExecute(op, val);\r\n\t    \t}\r\n\t\t    val.top(result);\r\n\r\n\t\t    return 0;\r\n\t    }\r\n\r\n\r\n    int Precedence(char oper)\r\n    {\r\n    \tif (     oper==&#39;*&#39; || oper==&#39;/&#39;)\r\n    \t\treturn 2;\r\n    \telse if( oper==&#39;+&#39; || oper==&#39;-&#39;)\r\n    \t\treturn 1;\r\n    \telse\r\n\t    \treturn 0;\r\n    }\r\n\r\n    void Execute(stack&lt;char&gt;&amp; op, stack&lt;int&gt;&amp; val)\r\n    {\r\n\t    switch (op)\r\n\t    {\r\n\t    \tcase &#39;+&#39;:\r\n\t    \t\tresult = val + val; //Unsure how to get results\r\n\t\t    \tbreak;\r\n\t\t\r\n\t    \tcase &#39;-&#39;:\r\n\t    \t\tresult = val - val; //Unsure how to get results\r\n\t\t    \tbreak;\r\n\t\t\t\r\n\t\t    case &#39;*&#39;:\r\n\t\t    \tresult = val * val; //Unsure how to get results\r\n\t\t    \tbreak;\r\n\t\t\t\r\n\t\t    case &#39;/&#39;:\r\n\t\t    \tresult = val / val; //Unsure how to get results\r\n\t\t    \tbreak;\r\n\t    }\r\n\t\r\n\t    return result;\r\n\t\r\n    }\r\n\r\n\r\n  [1]: http://gyazo.com/b49244efc88142c3cae928b1f4455fb1",
            "link": "https://stackoverflow.com/questions/49932274/c-carrano-infix-calculator-with-stacks",
            "title": "C++ Carrano infix calculator with stacks",
            "body": "<p>I'm working on a infix calculator for one of my classes and I'm fairly new to stacks and such and I have hit a wall when its comes to implementing the execution of the operations of the values. The function used for executing only receives a operator and one value parameter, so I'm lost as to how to perform the operations on a value when I need two. I am basing my code off of pseudo-code that was supplied by the professor (<a href=\"http://gyazo.com/b49244efc88142c3cae928b1f4455fb1\" rel=\"nofollow noreferrer\">This is a picture of it</a>). If anyone could point me in the right direction I'd appreciate it. This is what I have.</p>\n\n<pre><code>int main()\n{\n    stack&lt;char&gt; op;\n    stack&lt;int&gt;  val;\n\n        int result;\n        char ch;\n\n        cin &gt;&gt; ch;\n\n        while (ch != 'Q') //Once it reads 'Q' the program quits\n        {\n            switch (ch)\n            {\n                case '0': case '1': case '2': case '3': case '4':\n                case '5': case '6': case '7': case '8': case '9':\n                    val.push(ch);\n                    break;\n\n                case '(':\n                    op.push(ch);\n                    break;\n\n                case '+': case '-': case '*': case '/':\n                    if (op.empty())\n                    {\n                        op.push(ch);\n                    }\n                    else if (Precedence(ch) &gt; Precedence(op.top()))\n                    {\n                        op.push(ch);\n                    }\n                    else\n                    {\n                        while ( (!op.empty()) &amp;&amp; (Precedence(ch) &lt;=         Precedence(op.top())) )\n                        {\n                            Execute(op, val);\n                        }\n                        op.push(ch);\n                    }\n                    break;\n\n                case ')':\n                {\n                    while (op.top() != '(')\n                    {\n                        Execute(op, val);\n                    }\n                    op.pop();\n                    break;\n                }\n            }\n\n            cin &gt;&gt; ch;\n        }\n\n        while (!op.empty())\n        {\n            Execute(op, val);\n        }\n        val.top(result);\n\n        return 0;\n    }\n\n\nint Precedence(char oper)\n{\n    if (     oper=='*' || oper=='/')\n        return 2;\n    else if( oper=='+' || oper=='-')\n        return 1;\n    else\n        return 0;\n}\n\nvoid Execute(stack&lt;char&gt;&amp; op, stack&lt;int&gt;&amp; val)\n{\n    switch (op)\n    {\n        case '+':\n            result = val + val; //Unsure how to get results\n            break;\n\n        case '-':\n            result = val - val; //Unsure how to get results\n            break;\n\n        case '*':\n            result = val * val; //Unsure how to get results\n            break;\n\n        case '/':\n            result = val / val; //Unsure how to get results\n            break;\n    }\n\n    return result;\n\n}\n</code></pre>\n"
        }
    ],
    "has_more": true,
    "quota_max": 300,
    "quota_remaining": 189
}